// @dhis2-ui/help/build/cjs/help.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TextOverflow = exports.StatusError = exports.StatusValid = exports.StatusWarning = exports.Default = exports.default = void 0;

var _uiConstants = require("@dhis2/ui-constants");

var _react = _interopRequireDefault(require("react"));

var _help = require("./help.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const description = `
Small text for giving guiding information or feedback, especially for data entry instructions or form validation feedback.

\`\`\`js
import { Help } from '@dhis2/ui'
\`\`\`
`;
var _default = {
  title: 'Forms/Help',
  component: _help.Help,
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  },
  argTypes: {
    valid: { ..._uiConstants.sharedPropTypes.statusArgType
    },
    warning: { ..._uiConstants.sharedPropTypes.statusArgType
    },
    error: { ..._uiConstants.sharedPropTypes.statusArgType
    }
  },
  args: {
    children: 'Allow me to be of assistance'
  }
};
exports.default = _default;

const Template = args => /*#__PURE__*/_react.default.createElement(_help.Help, args);

const Default = Template.bind({});
exports.Default = Default;
const StatusWarning = Template.bind({});
exports.StatusWarning = StatusWarning;
StatusWarning.args = {
  warning: true
};
StatusWarning.storyName = 'Status: Warning';
const StatusValid = Template.bind({});
exports.StatusValid = StatusValid;
StatusValid.args = {
  valid: true
};
StatusValid.storyName = 'Status: Valid';
const StatusError = Template.bind({});
exports.StatusError = StatusError;
StatusError.args = {
  error: true
};
StatusError.storyName = 'Status: Error';

const TextOverflow = args => /*#__PURE__*/_react.default.createElement("div", {
  style: {
    width: 200
  }
}, /*#__PURE__*/_react.default.createElement(_help.Help, args, "I take up more space than my container"));

exports.TextOverflow = TextOverflow;
