// @dhis2-ui/switch/build/cjs/switch/switch.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Switch = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _uiConstants = require("@dhis2/ui-constants");

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _switchIcons = require("./switch-icons.js");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

class Switch extends _react.Component {
  constructor(...args) {
    super(...args);

    _defineProperty(this, "ref", /*#__PURE__*/(0, _react.createRef)());

    _defineProperty(this, "handleChange", e => {
      if (this.props.onChange) {
        this.props.onChange(this.createHandlerPayload(), e);
      }
    });

    _defineProperty(this, "handleBlur", e => {
      if (this.props.onBlur) {
        this.props.onBlur(this.createHandlerPayload(), e);
      }
    });

    _defineProperty(this, "handleFocus", e => {
      if (this.props.onFocus) {
        this.props.onFocus(this.createHandlerPayload(), e);
      }
    });
  }

  componentDidMount() {
    if (this.props.initialFocus) {
      this.ref.current.focus();
    }
  }

  createHandlerPayload() {
    return {
      value: this.props.value,
      name: this.props.name,
      checked: !this.props.checked
    };
  }

  render() {
    const {
      ariaLabel,
      checked,
      className,
      disabled,
      error,
      label,
      name,
      tabIndex,
      valid,
      value,
      warning,
      dense,
      dataTest,
      role
    } = this.props;
    const classes = (0, _classnames.default)({
      checked,
      disabled,
      valid,
      error,
      warning,
      dense
    });
    return /*#__PURE__*/_react.default.createElement("label", {
      "data-test": dataTest,
      className: _style.default.dynamic([["4271753770", [_uiConstants.colors.grey900, _uiConstants.theme.disabled, _uiConstants.theme.focus]]]) + " " + ((0, _classnames.default)(className, {
        disabled,
        dense
      }) || "")
    }, /*#__PURE__*/_react.default.createElement("input", {
      "aria-label": ariaLabel,
      type: "checkbox",
      role: role,
      ref: this.ref,
      name: name,
      value: value,
      checked: checked,
      disabled: disabled,
      tabIndex: tabIndex,
      onChange: this.handleChange,
      onFocus: this.handleFocus,
      onBlur: this.handleBlur,
      className: _style.default.dynamic([["4271753770", [_uiConstants.colors.grey900, _uiConstants.theme.disabled, _uiConstants.theme.focus]]])
    }), /*#__PURE__*/_react.default.createElement("div", {
      className: _style.default.dynamic([["4271753770", [_uiConstants.colors.grey900, _uiConstants.theme.disabled, _uiConstants.theme.focus]]]) + " " + ((0, _classnames.default)('icon', {
        dense
      }) || "")
    }, /*#__PURE__*/_react.default.createElement(_switchIcons.SwitchRegular, {
      className: classes
    })), label, /*#__PURE__*/_react.default.createElement(_style.default, {
      id: "4271753770",
      dynamic: [_uiConstants.colors.grey900, _uiConstants.theme.disabled, _uiConstants.theme.focus]
    }, [`label.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;color:${_uiConstants.colors.grey900};font-size:16px;line-height:20px;}`, "label.dense.__jsx-style-dynamic-selector{font-size:14px;line-height:16px;}", `label.disabled.__jsx-style-dynamic-selector{cursor:not-allowed;color:${_uiConstants.theme.disabled};}`, "input.__jsx-style-dynamic-selector{opacity:0;position:absolute;height:18px;width:35px;margin-left:3px;}", "label.dense.__jsx-style-dynamic-selector input.__jsx-style-dynamic-selector{height:14px;width:27px;}", ".icon.__jsx-style-dynamic-selector{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;margin-right:5px;border:2px solid transparent;padding:1px;border-radius:14px;}", "label.dense.__jsx-style-dynamic-selector .icon.__jsx-style-dynamic-selector{margin-right:3px;border-radius:12px;}", `input.__jsx-style-dynamic-selector:focus+.icon.__jsx-style-dynamic-selector{outline:3px solid ${_uiConstants.theme.focus};outline-offset:-1px;}`]));
  }

}

exports.Switch = Switch;
Switch.defaultProps = {
  checked: false,
  dataTest: 'dhis2-uicore-switch',
  role: 'switch'
};
Switch.propTypes = {
  /** Sets an aria-label attribute on the input */
  ariaLabel: _propTypes.default.string,
  checked: _propTypes.default.bool,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string,

  /** Makes the switch smaller for information-dense layouts */
  dense: _propTypes.default.bool,

  /** Disables the switch */
  disabled: _propTypes.default.bool,

  /** Applies 'error' styles for validation feedback. Mutually exclusive with `valid` and `warning` prop types */
  error: _uiConstants.sharedPropTypes.statusPropType,

  /** Grab initial focus on the page */
  initialFocus: _propTypes.default.bool,

  /** Label for the switch. Can be a string or an element, for example an image */
  label: _propTypes.default.node,

  /** Name associated with the switch. Passed to event handlers in object */
  name: _propTypes.default.string,

  /** Sets a role attribute on the input */
  role: _propTypes.default.string,
  tabIndex: _propTypes.default.string,

  /** Applies 'valid' styles for validation feedback. Mutually exclusive with `error` and `warning` prop types */
  valid: _uiConstants.sharedPropTypes.statusPropType,

  /** Value associated with the switch. Passed to event handlers in object */
  value: _propTypes.default.string,

  /** Applies 'warning' styles for validation feedback. Mutually exclusive with `valid` and `error` prop types */
  warning: _uiConstants.sharedPropTypes.statusPropType,

  /** Called with signature `({ name: string, value: string, checked: bool }, event)` */
  onBlur: _propTypes.default.func,

  /** Called with signature `({ name: string, value: string, checked: bool }, event)` */
  onChange: _propTypes.default.func,

  /** Called with signature `({ name: string, value: string, checked: bool }, event)` */
  onFocus: _propTypes.default.func
};
