// @dhis2-ui/logo/build/cjs/logo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LogoWhite = exports.Logo = exports.LogoIconWhite = exports.LogoIcon = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _logoIconSvg = require("./logo-icon-svg");

var _logoSvg = require("./logo-svg");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/*
 * This should likely not live in ui-core, but in ui-widgets
 */

/*
 * These are official colors for dhis2 logos.
 *
 * They are isolated here and not in @dhis2/ui-constants as they should not be
 * shared with any other components.
 *
 * https://github.com/dhis2/dhis2-identity
 *
 */
const blue = '#0080d4';
const white = '#ffffff';
const dark = '#212225';

const LogoIcon = ({
  className,
  dataTest
}) => /*#__PURE__*/_react.default.createElement(_logoIconSvg.LogoIconSvg, {
  iconColor: blue,
  className: className,
  dataTest: dataTest
});

exports.LogoIcon = LogoIcon;
LogoIcon.defaultProps = {
  dataTest: 'dhis2-uicore-logoicon'
};
LogoIcon.propTypes = {
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string
};

const LogoIconWhite = ({
  className,
  dataTest
}) => /*#__PURE__*/_react.default.createElement(_logoIconSvg.LogoIconSvg, {
  iconColor: white,
  className: className,
  dataTest: dataTest
});

exports.LogoIconWhite = LogoIconWhite;
LogoIconWhite.defaultProps = {
  dataTest: 'dhis2-uicore-logoiconwhite'
};
LogoIconWhite.propTypes = {
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string
};

const Logo = ({
  className,
  dataTest
}) => /*#__PURE__*/_react.default.createElement(_logoSvg.LogoSvg, {
  iconColor: blue,
  textColor: dark,
  className: className,
  dataTest: dataTest
});

exports.Logo = Logo;
Logo.defaultProps = {
  dataTest: 'dhis2-uicore-logo'
};
Logo.propTypes = {
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string
};

const LogoWhite = ({
  className,
  dataTest
}) => /*#__PURE__*/_react.default.createElement(_logoSvg.LogoSvg, {
  iconColor: white,
  textColor: white,
  className: className,
  dataTest: dataTest
});

exports.LogoWhite = LogoWhite;
LogoWhite.defaultProps = {
  dataTest: 'dhis2-uicore-logowhite'
};
LogoWhite.propTypes = {
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string
};
