// @dhis2-ui/modal/build/cjs/modal-content/modal-content.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ModalContent = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _propTypes = _interopRequireDefault(require("@dhis2/prop-types"));

var _uiConstants = require("@dhis2/ui-constants");

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const ModalContent = ({
  children,
  className,
  dataTest
}) => /*#__PURE__*/_react.default.createElement("div", {
  "data-test": dataTest,
  className: _style.default.dynamic([["936713121", [_uiConstants.spacers.dp24, _uiConstants.spacers.dp24]]]) + " " + (className || "")
}, children, /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "936713121",
  dynamic: [_uiConstants.spacers.dp24, _uiConstants.spacers.dp24]
}, [`div.__jsx-style-dynamic-selector{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;margin:${_uiConstants.spacers.dp24} 0;padding:0 ${_uiConstants.spacers.dp24};overflow-y:auto;}`]));

exports.ModalContent = ModalContent;
ModalContent.defaultProps = {
  dataTest: 'dhis2-uicore-modalcontent'
};
ModalContent.propTypes = {
  children: _propTypes.default.node,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string
};
