// @dhis2-ui/menu/build/cjs/flyout-menu/flyout-menu.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FlyoutMenu = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _card = require("@dhis2-ui/card");

var _uiConstants = require("@dhis2/ui-constants");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _index = require("../index.js");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const FlyoutMenu = ({
  children,
  className,
  dataTest,
  dense,
  maxHeight,
  maxWidth
}) => {
  const [openedSubMenu, setOpenedSubMenu] = (0, _react.useState)(null);

  const toggleSubMenu = index => {
    const toggleValue = index === openedSubMenu ? null : index;
    setOpenedSubMenu(toggleValue);
  };

  const cardCSS = {
    styles: /*#__PURE__*/_react.default.createElement(_style.default, {
      id: "1140426988",
      dynamic: [_uiConstants.spacers.dp4, maxHeight]
    }, [`.__jsx-style-dynamic-selector{padding:${_uiConstants.spacers.dp4} 0;max-height:${maxHeight};overflow-y:auto;}`]),
    className: _style.default.dynamic([["1140426988", [_uiConstants.spacers.dp4, maxHeight]]])
  };
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": dataTest,
    className: _style.default.dynamic([["2253458022", [dense ? '128' : '180', maxWidth, maxHeight]]]) + " " + (className || "")
  }, /*#__PURE__*/_react.default.createElement(_card.Card, {
    className: cardCSS.className
  }, /*#__PURE__*/_react.default.createElement(_index.Menu, {
    dense: dense
  }, _react.Children.map(children, (child, index) => /*#__PURE__*/(0, _react.isValidElement)(child) ? /*#__PURE__*/(0, _react.cloneElement)(child, {
    showSubMenu: openedSubMenu === index,
    toggleSubMenu: toggleSubMenu.bind(void 0, index)
  }) : child))), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "2253458022",
    dynamic: [dense ? '128' : '180', maxWidth, maxHeight]
  }, [`div.__jsx-style-dynamic-selector{display:inline-block;min-width:${dense ? '128' : '180'}px;max-width:${maxWidth};max-height:${maxHeight};}`]), cardCSS.styles);
};

exports.FlyoutMenu = FlyoutMenu;
FlyoutMenu.defaultProps = {
  dataTest: 'dhis2-uicore-menu',
  maxWidth: '380px',
  maxHeight: 'auto'
};
FlyoutMenu.propTypes = {
  /** Typically, but not limited to, `MenuItem` components */
  children: _propTypes.default.node,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string,

  /** Menu uses smaller dimensions */
  dense: _propTypes.default.bool,
  maxHeight: _propTypes.default.string,
  maxWidth: _propTypes.default.string
};
