// @dhis2-ui/modal/build/cjs/modal-actions/modal-actions.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ModalActions = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _propTypes = _interopRequireDefault(require("@dhis2/prop-types"));

var _uiConstants = require("@dhis2/ui-constants");

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const ModalActions = ({
  children,
  dataTest
}) => /*#__PURE__*/_react.default.createElement("div", {
  "data-test": dataTest,
  className: _style.default.dynamic([["1459377348", [_uiConstants.spacers.dp24, _uiConstants.spacers.dp24, _uiConstants.spacers.dp24]]])
}, children, /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "1459377348",
  dynamic: [_uiConstants.spacers.dp24, _uiConstants.spacers.dp24, _uiConstants.spacers.dp24]
}, [`div.__jsx-style-dynamic-selector{padding:0 ${_uiConstants.spacers.dp24} ${_uiConstants.spacers.dp24} ${_uiConstants.spacers.dp24};display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:end;-webkit-justify-content:flex-end;-ms-flex-pack:end;justify-content:flex-end;}`]));

exports.ModalActions = ModalActions;
ModalActions.defaultProps = {
  dataTest: 'dhis2-uicore-modalactions'
};
ModalActions.propTypes = {
  children: _propTypes.default.node,
  dataTest: _propTypes.default.string
};
