// @dhis2-ui/tab/build/cjs/tab-bar/tab-bar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TabBar = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _scrollBar = require("./scroll-bar.js");

var _tabs = require("./tabs.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const TabBar = ({
  fixed,
  children,
  className,
  scrollable,
  dataTest
}) => {
  if (scrollable) {
    return /*#__PURE__*/_react.default.createElement("div", {
      className: className,
      "data-test": dataTest
    }, /*#__PURE__*/_react.default.createElement(_scrollBar.ScrollBar, {
      dataTest: `${dataTest}-scrollbar`
    }, /*#__PURE__*/_react.default.createElement(_tabs.Tabs, {
      fixed: fixed,
      dataTest: `${dataTest}-tabs`
    }, children)));
  }

  return /*#__PURE__*/_react.default.createElement("div", {
    className: className,
    "data-test": dataTest
  }, /*#__PURE__*/_react.default.createElement(_tabs.Tabs, {
    fixed: fixed,
    dataTest: `${dataTest}-tabs`
  }, children));
};

exports.TabBar = TabBar;
TabBar.defaultProps = {
  dataTest: 'dhis2-uicore-tabbar'
};
TabBar.propTypes = {
  children: _propTypes.default.node,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string,

  /** Fixed tabs fill the width of their container. If false (i.e. fluid), tabs take up an amount of space defined by the tab name. Fluid tabs should be used most of the time. */
  fixed: _propTypes.default.bool,

  /** Enables horizontal scrolling for many tabs that don't fit the width of the container */
  scrollable: _propTypes.default.bool
};
