// @dhis2-ui/cover/build/cjs/cover.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Cover = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _uiConstants = require("@dhis2/ui-constants");

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const createClickHandler = onClick => event => {
  // don't respond to clicks that originated in the children
  if (onClick && event.target === event.currentTarget) {
    onClick({}, event);
  }
};

const Cover = ({
  children,
  className,
  dataTest,
  onClick,
  translucent
}) => /*#__PURE__*/_react.default.createElement("div", {
  onClick: createClickHandler(onClick),
  "data-test": dataTest,
  className: _style.default.dynamic([["3798453419", [_uiConstants.layers.applicationTop]]]) + " " + ((0, _classnames.default)(className, {
    translucent
  }) || "")
}, children, /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "3798453419",
  dynamic: [_uiConstants.layers.applicationTop]
}, [`div.__jsx-style-dynamic-selector{position:absolute;top:0;right:0;bottom:0;left:0;z-index:${_uiConstants.layers.applicationTop};}`, "div.translucent.__jsx-style-dynamic-selector{background:rgba(33,43,54,0.4);}"]));

exports.Cover = Cover;
Cover.defaultProps = {
  dataTest: 'dhis2-uicore-componentcover'
};
Cover.propTypes = {
  children: _propTypes.default.node,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string,

  /** Adds a semi-transparent background to the cover */
  translucent: _propTypes.default.bool,
  onClick: _propTypes.default.func
};
