// @dhis2-ui/cover/build/cjs/cover.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WithCenteredContentCircularLoader = exports.WithClickHandler = exports.Translucent = exports.Default = exports.default = void 0;

var _center = require("@dhis2-ui/center");

var _loader = require("@dhis2-ui/loader");

var _react = _interopRequireDefault(require("react"));

var _cover = require("./cover.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const description = `
Covers sibling components.  Useful for covering a component while it is loading.

\`\`\`js
import { Cover } from '@dhis2/ui'
\`\`\`
`;
var _default = {
  title: 'Helpers/Component Cover',
  component: _cover.Cover,
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  }
};
exports.default = _default;

const Template = args => /*#__PURE__*/_react.default.createElement("div", {
  style: {
    width: '400px',
    height: '200px',
    position: 'relative',
    border: '1px dotted grey'
  }
}, /*#__PURE__*/_react.default.createElement(_cover.Cover, args), /*#__PURE__*/_react.default.createElement("h1", null, "Text behind the cover"), /*#__PURE__*/_react.default.createElement("p", null, "Lorem ipsum"));

const Default = Template.bind({});
exports.Default = Default;
const Translucent = Template.bind({});
exports.Translucent = Translucent;
Translucent.args = {
  translucent: true
};
const WithClickHandler = Template.bind({});
exports.WithClickHandler = WithClickHandler;
WithClickHandler.args = {
  onClick: () => alert('Cover was clicked')
};
const WithCenteredContentCircularLoader = Template.bind({});
exports.WithCenteredContentCircularLoader = WithCenteredContentCircularLoader;
WithCenteredContentCircularLoader.args = {
  translucent: true,
  children: /*#__PURE__*/_react.default.createElement(_center.Center, null, /*#__PURE__*/_react.default.createElement(_loader.CircularLoader, null))
};
