// @dhis2-ui/menu/build/cjs/menu-divider/menu-divider.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MenuDivider = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _divider = require("@dhis2-ui/divider");

var _uiConstants = require("@dhis2/ui-constants");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const MenuDivider = ({
  className,
  dataTest,
  dense
}) => /*#__PURE__*/_react.default.createElement("li", {
  "data-test": dataTest,
  className: _style.default.dynamic([["591815244", [_uiConstants.colors.white]]]) + " " + (className || "")
}, /*#__PURE__*/_react.default.createElement(_divider.Divider, {
  dense: dense
}), /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "591815244",
  dynamic: [_uiConstants.colors.white]
}, [`li.__jsx-style-dynamic-selector{list-style:none;background-color:${_uiConstants.colors.white};-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;padding:0;line-height:0;}`]));

exports.MenuDivider = MenuDivider;
MenuDivider.defaultProps = {
  dataTest: 'dhis2-uicore-menudivider'
};
MenuDivider.propTypes = {
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string,
  dense: _propTypes.default.bool
};
