// @dhis2-ui/layer/build/cjs/layer.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Layer = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _portal = require("@dhis2-ui/portal");

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const createClickHandler = onClick => event => {
  // don't respond to clicks that originated in the children
  if (onClick && event.target === event.currentTarget) {
    onClick({}, event);
  }
};

const Layer = ({
  children,
  className,
  dataTest,
  disablePortal,
  level,
  onClick,
  position,
  translucent
}) => /*#__PURE__*/_react.default.createElement(_portal.Portal, {
  disable: disablePortal
}, /*#__PURE__*/_react.default.createElement("div", {
  "data-test": dataTest,
  onClick: createClickHandler(onClick),
  className: "jsx-2944291715 " + _style.default.dynamic([["2635812695", [level]]]) + " " + ((0, _classnames.default)('layer', className, position, {
    translucent
  }) || "")
}, children, /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "2635812695",
  dynamic: [level]
}, [`div.__jsx-style-dynamic-selector{z-index:${level};}`]), /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "2944291715"
}, ["div.jsx-2944291715{top:0;left:0;min-height:100vh;min-width:100vw;}", "div.fixed.jsx-2944291715{position:fixed;height:100vh;width:100vw;}", "div.absolute.jsx-2944291715{position:absolute;height:100%;width:100%;}", "div.translucent.jsx-2944291715{background-color:rgba(33,43,54,0.4);}"])));

exports.Layer = Layer;
Layer.defaultProps = {
  position: 'fixed',
  dataTest: 'dhis2-uicore-layer',
  level: 'auto'
};
Layer.propTypes = {
  children: _propTypes.default.node,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string,

  /** Disable the Portal, useful for nesting layers */
  disablePortal: _propTypes.default.bool,

  /** Z-index level */
  level: _propTypes.default.oneOfType([_propTypes.default.number, _propTypes.default.string]),
  position: _propTypes.default.oneOf(['absolute', 'fixed']),

  /** Adds a semi-transparent background */
  translucent: _propTypes.default.bool,

  /** Click handler */
  onClick: _propTypes.default.func
};
