// @dhis2-ui/layer/build/cjs/layer.stories.e2e.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NestedHigherLevelEndsOnTop = exports.LevelsAreRespectedWhenNesting = exports.NestedHigherLevels = exports.NestedLowerLevels = exports.InequalSiblings = exports.EqualSiblings = exports.WithClickHandler = exports.Blocking = exports.Default = exports.default = void 0;

var _uiConstants = require("@dhis2/ui-constants");

var _react = _interopRequireDefault(require("react"));

var _layer = require("./layer.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

window.onButtonClick = window.Cypress && window.Cypress.cy.stub();
window.onLayerClick = window.Cypress && window.Cypress.cy.stub();

const createNamedLayerClick = name => () => {
  window.onLayerClick(name);
};

var _default = {
  title: 'Layer',
  component: _layer.Layer
};
exports.default = _default;

const Default = () => /*#__PURE__*/_react.default.createElement(_layer.Layer, null, /*#__PURE__*/_react.default.createElement("p", null, "I am a child"));

exports.Default = Default;

const Blocking = () => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("button", {
  onClick: window.onButtonClick
}, "Test"), /*#__PURE__*/_react.default.createElement(_layer.Layer, null));

exports.Blocking = Blocking;

const WithClickHandler = () => /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  onClick: window.onLayerClick
}, /*#__PURE__*/_react.default.createElement("button", {
  onClick: window.onButtonClick
}, "Test"));

exports.WithClickHandler = WithClickHandler;

const EqualSiblings = () => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  onClick: createNamedLayerClick('first')
}), /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  onClick: createNamedLayerClick('second')
}));

exports.EqualSiblings = EqualSiblings;

const InequalSiblings = () => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  level: _uiConstants.layers.alert,
  onClick: createNamedLayerClick('alert')
}), /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  level: _uiConstants.layers.blocking,
  onClick: createNamedLayerClick('blocking')
}), /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  level: _uiConstants.layers.applicationTop,
  onClick: createNamedLayerClick('applicationTop')
}));

exports.InequalSiblings = InequalSiblings;

const NestedLowerLevels = () => /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  level: _uiConstants.layers.alert,
  dataTest: "alert",
  onClick: createNamedLayerClick('alert')
}, /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  level: _uiConstants.layers.blocking,
  dataTest: "blocking",
  disablePortal: true,
  onClick: createNamedLayerClick('blocking')
}));

exports.NestedLowerLevels = NestedLowerLevels;

const NestedHigherLevels = () => /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  level: _uiConstants.layers.blocking,
  dataTest: "blocking",
  onClick: createNamedLayerClick('blocking')
}, /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  level: _uiConstants.layers.alert,
  dataTest: "alert",
  onClick: createNamedLayerClick('alert')
}));

exports.NestedHigherLevels = NestedHigherLevels;

const LevelsAreRespectedWhenNesting = () => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  level: 1000
}, /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  level: 1000
}, /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  level: 1000,
  onClick: createNamedLayerClick('1000')
}))), /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  level: 1001,
  onClick: createNamedLayerClick('1001')
}));

exports.LevelsAreRespectedWhenNesting = LevelsAreRespectedWhenNesting;

const NestedHigherLevelEndsOnTop = () => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  level: _uiConstants.layers.applicationTop
}, /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  level: _uiConstants.layers.alert,
  onClick: createNamedLayerClick('alert')
})), /*#__PURE__*/_react.default.createElement(_layer.Layer, {
  level: _uiConstants.layers.blocking,
  onClick: createNamedLayerClick('blocking')
}));

exports.NestedHigherLevelEndsOnTop = NestedHigherLevelEndsOnTop;
