// @dhis2-ui/field/build/cjs/field/field.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Default = exports.default = void 0;

var _input = require("@dhis2-ui/input");

var _uiConstants = require("@dhis2/ui-constants");

var _react = _interopRequireDefault(require("react"));

var _field = require("./field.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const description = `
A useful container for form components, including a label, help text, and validation text.

\`\`\`js
import { Field } from '@dhis2/ui'
\`\`\`
`;
var _default = {
  title: 'Forms/Field',
  component: _field.Field,
  argTypes: {
    valid: { ..._uiConstants.sharedPropTypes.statusArgType
    },
    warning: { ..._uiConstants.sharedPropTypes.statusArgType
    },
    error: { ..._uiConstants.sharedPropTypes.statusArgType
    }
  },
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  }
};
exports.default = _default;

const Default = args => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_field.Field, args, /*#__PURE__*/_react.default.createElement(_input.Input, {
  onChange: () => {
    console.log('Nothing happens');
  },
  name: "input",
  label: "An input"
})), /*#__PURE__*/_react.default.createElement(_field.Field, {
  helpText: "Help!",
  label: "Second field"
}, /*#__PURE__*/_react.default.createElement(_input.Input, {
  onChange: () => {
    console.log('Nothing happens');
  },
  name: "input2",
  label: "An second input"
})));

exports.Default = Default;
Default.args = {
  label: 'First field (change me with controls)',
  helpText: 'Help text!',
  validationText: "I'm validation text"
};
