// @dhis2-ui/card/build/cjs/card.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Card = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _uiConstants = require("@dhis2/ui-constants");

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const Card = ({
  className,
  children,
  dataTest
}) => /*#__PURE__*/_react.default.createElement("div", {
  "data-test": dataTest,
  className: _style.default.dynamic([["166161216", [_uiConstants.colors.white]]]) + " " + ((0, _classnames.default)(className) || "")
}, children, /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "166161216",
  dynamic: [_uiConstants.colors.white]
}, [`div.__jsx-style-dynamic-selector{display:inline-block;position:relative;width:100%;height:100%;border-radius:3px;background:${_uiConstants.colors.white};box-shadow:0 0 1px 0 rgba(64,75,90,0.2), 0 2px 1px 0 rgba(64,75,90,0.28);}`]));

exports.Card = Card;
Card.defaultProps = {
  dataTest: 'dhis2-uicore-card'
};
Card.propTypes = {
  children: _propTypes.default.node,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string
};
