// @dhis2-ui/select/build/es/single-select-option/single-select-option.js
import _JSXStyle from "styled-jsx/style";
import propTypes from '@dhis2/prop-types';
import { colors, spacers } from '@dhis2/ui-constants';
import cx from 'classnames';
import React from 'react';

const SingleSelectOption = ({
  label,
  active,
  disabled,
  onClick,
  className,
  dataTest,
  value
}) => /*#__PURE__*/React.createElement("div", {
  onClick: e => onClick({}, e),
  "data-test": dataTest,
  "data-value": value,
  "data-label": label,
  className: _JSXStyle.dynamic([["2046100779", [colors.grey900, spacers.dp8, spacers.dp12, colors.grey200, colors.teal700, colors.white, colors.grey500]]]) + " " + (cx(className, {
    disabled,
    active
  }) || "")
}, label, /*#__PURE__*/React.createElement(_JSXStyle, {
  id: "2046100779",
  dynamic: [colors.grey900, spacers.dp8, spacers.dp12, colors.grey200, colors.teal700, colors.white, colors.grey500]
}, [`div.__jsx-style-dynamic-selector{cursor:pointer;font-size:14px;-webkit-text-decoration:none;text-decoration:none;color:${colors.grey900};padding:${spacers.dp8} ${spacers.dp12};}`, `div.__jsx-style-dynamic-selector:hover{background-color:${colors.grey200};}`, `div.__jsx-style-dynamic-selector:active,div.active.__jsx-style-dynamic-selector{background-color:${colors.teal700};color:${colors.white};cursor:auto;}`, `div.disabled.__jsx-style-dynamic-selector{color:${colors.grey500};cursor:not-allowed;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;}`, "div.disabled.__jsx-style-dynamic-selector:hover{background-color:initial;}"]));

SingleSelectOption.defaultProps = {
  dataTest: 'dhis2-uicore-singleselectoption'
};
SingleSelectOption.propTypes = {
  label: propTypes.string.isRequired,
  value: propTypes.string.isRequired,
  active: propTypes.bool,
  className: propTypes.string,
  dataTest: propTypes.string,
  disabled: propTypes.bool,
  onClick: propTypes.func
};
export { SingleSelectOption };
