// @dhis2-ui/popper/build/cjs/popper.stories.e2e.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.VirtualElementAsReference = exports.DOMNodeAsReference = exports.ReactRefAsReference = exports.LeftEnd = exports.LeftStart = exports.Left = exports.RightEnd = exports.RightStart = exports.Right = exports.BottomEnd = exports.BottomStart = exports.Bottom = exports.TopEnd = exports.TopStart = exports.Top = exports.default = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _propTypes = _interopRequireDefault(require("@dhis2/prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _popper = require("./popper.js");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const PopperPlacement = ({
  placement
}) => {
  const ref = (0, _react.useRef)();
  return /*#__PURE__*/_react.default.createElement("div", {
    className: "jsx-305551169" + " " + "box"
  }, /*#__PURE__*/_react.default.createElement("div", {
    ref: ref,
    className: "jsx-305551169" + " " + "reference-element"
  }, "Reference element"), /*#__PURE__*/_react.default.createElement(_popper.Popper, {
    reference: ref,
    placement: placement
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "jsx-305551169" + " " + "popper-content"
  }, "Popper")), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "305551169"
  }, [".box.jsx-305551169{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;width:400px;height:400px;margin-bottom:1000px;background-color:aliceblue;}", ".reference-element.jsx-305551169{width:100px;height:50px;background-color:cadetblue;text-align:center;padding:6px;}", ".popper-content.jsx-305551169{width:80px;height:30px;background-color:lightblue;text-align:center;padding:6px;}"]));
};

PopperPlacement.propTypes = {
  placement: _propTypes.default.string
};
var _default = {
  title: 'Popper',
  component: _popper.Popper
};
exports.default = _default;

const Top = () => /*#__PURE__*/_react.default.createElement(PopperPlacement, {
  placement: "top"
});

exports.Top = Top;

const TopStart = () => /*#__PURE__*/_react.default.createElement(PopperPlacement, {
  placement: "top-start"
});

exports.TopStart = TopStart;

const TopEnd = () => /*#__PURE__*/_react.default.createElement(PopperPlacement, {
  placement: "top-end"
});

exports.TopEnd = TopEnd;

const Bottom = () => /*#__PURE__*/_react.default.createElement(PopperPlacement, {
  placement: "bottom"
});

exports.Bottom = Bottom;

const BottomStart = () => /*#__PURE__*/_react.default.createElement(PopperPlacement, {
  placement: "bottom-start"
});

exports.BottomStart = BottomStart;

const BottomEnd = () => /*#__PURE__*/_react.default.createElement(PopperPlacement, {
  placement: "bottom-end"
});

exports.BottomEnd = BottomEnd;

const Right = () => /*#__PURE__*/_react.default.createElement(PopperPlacement, {
  placement: "right"
});

exports.Right = Right;

const RightStart = () => /*#__PURE__*/_react.default.createElement(PopperPlacement, {
  placement: "right-start"
});

exports.RightStart = RightStart;

const RightEnd = () => /*#__PURE__*/_react.default.createElement(PopperPlacement, {
  placement: "right-end"
});

exports.RightEnd = RightEnd;

const Left = () => /*#__PURE__*/_react.default.createElement(PopperPlacement, {
  placement: "left"
});

exports.Left = Left;

const LeftStart = () => /*#__PURE__*/_react.default.createElement(PopperPlacement, {
  placement: "left-start"
});

exports.LeftStart = LeftStart;

const LeftEnd = () => /*#__PURE__*/_react.default.createElement(PopperPlacement, {
  placement: "left-end"
});

exports.LeftEnd = LeftEnd;

const ReactRefAsReference = () => {
  const ref = (0, _react.useRef)();
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("button", {
    ref: ref
  }, "Reference"), /*#__PURE__*/_react.default.createElement(_popper.Popper, {
    placement: "bottom-start",
    reference: ref
  }, "Popper"));
};

exports.ReactRefAsReference = ReactRefAsReference;

const DOMNodeAsReference = () => {
  const [node, setNode] = (0, _react.useState)(null);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("button", {
    ref: setNode
  }, "Reference"), /*#__PURE__*/_react.default.createElement(_popper.Popper, {
    placement: "bottom-start",
    reference: node
  }, "Popper"));
};

exports.DOMNodeAsReference = DOMNodeAsReference;

const VirtualElementAsReference = () => {
  const virtualElement = {
    getBoundingClientRect: () => ({
      top: 200,
      left: 200,
      bottom: 'auto',
      right: 'auto',
      width: 0,
      height: 0
    })
  };
  return /*#__PURE__*/_react.default.createElement(_popper.Popper, {
    placement: "bottom-start",
    reference: virtualElement
  }, "Popper");
};

exports.VirtualElementAsReference = VirtualElementAsReference;
