// @dhis2-ui/legend/build/cjs/legend.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Required = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _legend = require("./legend.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const description = `
Legends are to be used in a Field Set to describe the form fields.  They may indicate that the fields are required.

See the [Field Set](../?path=/docs/forms-field-set-field-set--usage-example-a-radio-button-group-with-error-status) for a usage example.

\`\`\`js
import { Legend } from '@dhis2/ui'
\`\`\`
`;
var _default = {
  title: 'Forms/Field Set/Legend',
  component: _legend.Legend,
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  }
};
exports.default = _default;

const Template = args => /*#__PURE__*/_react.default.createElement(_legend.Legend, args, "I am wrapped in a legend which has some styling");

const Default = Template.bind({});
exports.Default = Default;
const Required = Template.bind({});
exports.Required = Required;
Required.args = {
  required: true
};
