// @dhis2-ui/legend/build/cjs/legend.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Legend = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _required = require("@dhis2-ui/required");

var _uiConstants = require("@dhis2/ui-constants");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const Legend = ({
  className,
  children,
  required,
  dataTest
}) => /*#__PURE__*/_react.default.createElement("legend", {
  "data-test": dataTest,
  className: _style.default.dynamic([["945530629", [_uiConstants.colors.grey900]]]) + " " + (className || "")
}, children, required && /*#__PURE__*/_react.default.createElement(_required.Required, {
  dataTest: `${dataTest}-required`
}), /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "945530629",
  dynamic: [_uiConstants.colors.grey900]
}, [`legend.__jsx-style-dynamic-selector{font-size:14px;line-height:16px;color:${_uiConstants.colors.grey900};}`]));

exports.Legend = Legend;
Legend.defaultProps = {
  dataTest: 'dhis2-uicore-legend'
};
Legend.propTypes = {
  children: _propTypes.default.node,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string,

  /** Indicates the associated field set is required */
  required: _propTypes.default.bool
};
