// @dhis2-ui/table/build/cjs/data-table/data-table.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InlineFiltering = exports.ColumnHeaderSorting = exports.ScrollingDataTableWithToolbars = exports.FixedHeaderAndTwoColumns = exports.FixedFirstColumn = exports.FixedHeader = exports.SelectableRows = exports.ExpandableContent = exports.Toolbars = exports.CellStyling = exports.DraggableRows = exports.LargeCells = exports.BorderedCells = exports.Default = exports.default = void 0;

var _uiCore = require("@dhis2/ui-core");

var _react = _interopRequireWildcard(require("react"));

var _dataTableBody = require("./data-table-body.js");

var _dataTableCell = require("./data-table-cell.js");

var _dataTableColumnHeader = require("./data-table-column-header/data-table-column-header.js");

var _dataTableFoot = require("./data-table-foot.js");

var _dataTableHead = require("./data-table-head.js");

var _dataTableRow = require("./data-table-row/data-table-row.js");

var _dataTableToolbar = require("./data-table-toolbar.js");

var _dataTable = require("./data-table.js");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const subtitle = 'Used to display information in a standard, effective way.';
const description = `
Should be used with multiple DataTable-related child components - see the datatable and examples below.

\`\`\`js
import {
    DataTable,
    DataTableToolbar,
    DataTableHead,
    DataTableBody,
    DataTableFoot,
    DataTableRow,
    DataTableCell,
    DataTableColumnHeader,
} from '@dhis2/ui'
\`\`\`
`;
var _default = {
  title: 'Data Display/DataTable',
  component: _dataTable.DataTable,
  // Add subcomponents to the args datatable
  subcomponents: {
    DataTableToolbar: _dataTableToolbar.DataTableToolbar,
    DataTableHead: _dataTableHead.DataTableHead,
    DataTableBody: _dataTableBody.DataTableBody,
    DataTableFoot: _dataTableFoot.DataTableFoot,
    DataTableRow: _dataTableRow.DataTableRow,
    DataTableCell: _dataTableCell.DataTableCell,
    DataTableColumnHeader: _dataTableColumnHeader.DataTableColumnHeader
  },
  parameters: {
    componentSubtitle: subtitle,
    docs: {
      description: {
        component: description
      }
    }
  },
  argTypes: {
    // hide control from `children` because that can't be live edited
    children: {
      control: {
        disable: true
      }
    },
    // hide child component controls completely to avoid confusion
    bordered: {
      datatable: {
        disable: true
      }
    },
    large: {
      datatable: {
        disable: true
      }
    },
    draggable: {
      datatable: {
        disable: true
      }
    }
  }
};
exports.default = _default;

const BasicTemplate = ({
  bordered,
  large,
  draggable,
  ...args
}) => /*#__PURE__*/_react.default.createElement(_dataTable.DataTable, args, /*#__PURE__*/_react.default.createElement(_dataTableHead.DataTableHead, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, draggable && /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  large: large
}), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  large: large
}, "First name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  large: large
}, "Last name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  large: large
}, "Incident date"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  large: large
}, "Last updated"))), /*#__PURE__*/_react.default.createElement(_dataTableBody.DataTableBody, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, {
  draggable: draggable
}, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  large: large,
  bordered: bordered
}, "Onyekachukwu"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  large: large,
  bordered: bordered
}, "Kariuki"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  large: large,
  bordered: bordered
}, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  large: large,
  bordered: bordered
}, "05/25/1972")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, {
  draggable: draggable
}, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  large: large,
  bordered: bordered
}, "Kwasi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  large: large,
  bordered: bordered
}, "Okafor"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  large: large,
  bordered: bordered
}, "08/11/2010"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  large: large,
  bordered: bordered
}, "02/26/1991")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, {
  draggable: draggable
}, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  large: large,
  bordered: bordered
}, "Siyabonga"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  large: large,
  bordered: bordered
}, "Abiodun"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  large: large,
  bordered: bordered
}, "07/21/1981"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  large: large,
  bordered: bordered
}, "02/06/2007"))), /*#__PURE__*/_react.default.createElement(_dataTableFoot.DataTableFoot, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  colSpan: draggable ? '5' : '4'
}, "Footer content"))));

const Default = BasicTemplate.bind({});
exports.Default = Default;
Default.args = {};
const BorderedCells = BasicTemplate.bind({});
exports.BorderedCells = BorderedCells;
BorderedCells.args = {
  bordered: true
};
const LargeCells = BasicTemplate.bind({});
exports.LargeCells = LargeCells;
LargeCells.args = {
  large: true
};
const DraggableRows = BasicTemplate.bind({});
exports.DraggableRows = DraggableRows;
DraggableRows.args = {
  draggable: true
};

const IndividualCellTemplate = args => /*#__PURE__*/_react.default.createElement(_dataTable.DataTable, args, /*#__PURE__*/_react.default.createElement(_dataTableHead.DataTableHead, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Header"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Header"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Header"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Header"))), /*#__PURE__*/_react.default.createElement(_dataTableBody.DataTableBody, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  tag: "th"
}, "Active Cells"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Inactive cell"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  active: true,
  onClick: () => {
    console.log('You clicked on a DataTableCell');
  }
}, "Acive Cell with onClick"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Inactive cell")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  tag: "th"
}, "Alignment"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  align: "left"
}, "Left"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  align: "center"
}, "Center"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  align: "right"
}, "Right")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  tag: "th"
}, "Built in formatting"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  error: true
}, "Error"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  valid: true
}, "Valid"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  muted: true
}, "Muted"))));

const CellStyling = IndividualCellTemplate.bind({});
exports.CellStyling = CellStyling;
CellStyling.args = {};

const ToolbarsTemplate = args => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_dataTableToolbar.DataTableToolbar, null, /*#__PURE__*/_react.default.createElement("p", null, "Content")), /*#__PURE__*/_react.default.createElement(_dataTable.DataTable, args, /*#__PURE__*/_react.default.createElement(_dataTableHead.DataTableHead, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "First name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Last name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Incident date"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Last updated"))), /*#__PURE__*/_react.default.createElement(_dataTableBody.DataTableBody, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Onyekachukwu"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Kariuki"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "05/25/1972")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Kwasi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Okafor"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "08/11/2010"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/26/1991")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Siyabonga"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Abiodun"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/21/1981"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/06/2007")))), /*#__PURE__*/_react.default.createElement(_dataTableToolbar.DataTableToolbar, {
  position: "bottom"
}, /*#__PURE__*/_react.default.createElement("p", null, "Content (bottom)")));

const Toolbars = ToolbarsTemplate.bind({});
exports.Toolbars = Toolbars;
Toolbars.args = {};

const ExpandableContentTemplate = args => {
  const [openRowIndex, setOpenRowIndex] = (0, _react.useState)(null);

  const toggleOpenRow = index => setOpenRowIndex(openRowIndex === index ? null : index);

  const style = {
    margin: 8,
    padding: 4,
    backgroundColor: 'lightblue'
  };

  const expandableContent = /*#__PURE__*/_react.default.createElement("div", {
    style: style
  }, "More info about this row!");

  return /*#__PURE__*/_react.default.createElement(_dataTable.DataTable, args, /*#__PURE__*/_react.default.createElement(_dataTableHead.DataTableHead, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "First name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Last name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Incident date"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Last updated"))), /*#__PURE__*/_react.default.createElement(_dataTableBody.DataTableBody, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, {
    expanded: openRowIndex === 0,
    onExpandToggle: payload => {
      console.log(payload);
      toggleOpenRow(0);
    },
    expandableContent: expandableContent
  }, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Onyekachukwu"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Kariuki"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "05/25/1972")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, {
    expanded: openRowIndex === 1,
    onExpandToggle: () => toggleOpenRow(1),
    expandableContent: expandableContent
  }, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Kwasi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Okafor"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "08/11/2010"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/26/1991")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, {
    expanded: openRowIndex === 2,
    onExpandToggle: () => toggleOpenRow(2),
    expandableContent: expandableContent
  }, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Siyabonga"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Abiodun"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/21/1981"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/06/2007"))));
};

const ExpandableContent = ExpandableContentTemplate.bind({});
exports.ExpandableContent = ExpandableContent;
ExpandableContent.args = {};

const SelectableRowsTemplate = args => {
  const [selected, setSelected] = (0, _react.useState)({
    id_2: true
  });

  const toggleSelected = ({
    value,
    checked
  }) => {
    setSelected({ ...selected,
      [value]: checked
    });
  };

  const toggleAll = ({
    checked
  }) => {
    setSelected({
      id_1: checked,
      id_2: checked,
      id_3: checked
    });
  };

  const allSelected = () => Object.values(selected).filter(value => value).length === 3;

  return /*#__PURE__*/_react.default.createElement(_dataTable.DataTable, args, /*#__PURE__*/_react.default.createElement(_dataTableHead.DataTableHead, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
    width: "48px"
  }, /*#__PURE__*/_react.default.createElement(_uiCore.Checkbox, {
    checked: allSelected(),
    onChange: toggleAll
  })), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "First name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Last name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Incident date"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Last updated"))), /*#__PURE__*/_react.default.createElement(_dataTableBody.DataTableBody, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, {
    selected: !!selected.id_1
  }, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
    width: "48px"
  }, /*#__PURE__*/_react.default.createElement(_uiCore.Checkbox, {
    value: "id_1",
    checked: !!selected.id_1,
    onChange: toggleSelected
  })), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Onyekachukwu"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Kariuki"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "05/25/1972")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, {
    selected: !!selected.id_2
  }, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
    width: "48px"
  }, /*#__PURE__*/_react.default.createElement(_uiCore.Checkbox, {
    value: "id_2",
    checked: !!selected.id_2,
    onChange: toggleSelected
  })), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Kwasi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Okafor"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "08/11/2010"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/26/1991")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, {
    selected: !!selected.id_3
  }, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
    width: "48px"
  }, /*#__PURE__*/_react.default.createElement(_uiCore.Checkbox, {
    value: "id_3",
    checked: !!selected.id_3,
    onChange: toggleSelected
  })), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Siyabonga"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Abiodun"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/21/1981"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/06/2007"))));
};

const SelectableRows = SelectableRowsTemplate.bind({});
exports.SelectableRows = SelectableRows;
SelectableRows.args = {};

const FixedHeaderTemplate = args => /*#__PURE__*/_react.default.createElement(_dataTable.DataTable, args, /*#__PURE__*/_react.default.createElement(_dataTableHead.DataTableHead, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "First name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Last name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Incident date"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Last updated"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Age"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Registering unit"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Assigned user"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Status"))), /*#__PURE__*/_react.default.createElement(_dataTableBody.DataTableBody, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Onyekachukwu"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Kariuki"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "05/25/1972"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "66"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Jawi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Sofie Hubert"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Kwasi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Okafor"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "08/11/2010"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/26/1991"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "38"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Mokassie MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Dashonte Clarke"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Siyabonga"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Abiodun"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/21/1981"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "98"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Bathurst MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Unassigned"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Chiyembekezo"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Okeke"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/23/1982"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/15/2003"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "2"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Mayolla MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Wan Gengxin"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Mtendere"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Afolayan"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "08/12/1994"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "05/12/1972"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "37"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Gbangadu MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Gvozden Boskovsky"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Inyene"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Okonkwo"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "04/01/1971"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "03/16/2000"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "70"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Kunike Barina"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Oscar de la Cavaller\xEDa"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Amaka"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Pretorius"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/25/1996"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "09/15/1986"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "32"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Bargbo"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Alberto Raya"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Meti"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Abiodun"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "10/24/2010"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/26/1989"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "8"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Majihun MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Unassigned"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Eshe"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Okeke"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/31/1995"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/31/1995"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "63"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Mambiama CHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Shadrias Pearson"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Obi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Okafor"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "06/07/1990"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/03/2006"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "28"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Dalakuru CHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Anatoliy Shcherbatykh"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete"))));

const FixedHeader = FixedHeaderTemplate.bind({});
exports.FixedHeader = FixedHeader;
FixedHeader.args = {
  scrollHeight: '350px'
};

const FixedFirstColumnTemplate = args => /*#__PURE__*/_react.default.createElement(_dataTable.DataTable, args, /*#__PURE__*/_react.default.createElement(_dataTableHead.DataTableHead, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0",
  left: "0"
}, "First name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Last name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Incident date"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Last updated"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Age"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Registering unit"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Assigned user"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, null, "Status"))), /*#__PURE__*/_react.default.createElement(_dataTableBody.DataTableBody, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0"
}, "Onyekachukwu"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Kariuki"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "05/25/1972"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "66"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Jawi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Sofie Hubert"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0"
}, "Kwasi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Okafor"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "08/11/2010"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/26/1991"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "38"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Mokassie MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Dashonte Clarke"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0"
}, "Siyabonga"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Abiodun"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/21/1981"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "98"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Bathurst MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Unassigned"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0"
}, "Chiyembekezo"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Okeke"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/23/1982"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/15/2003"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "2"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Mayolla MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Wan Gengxin"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0"
}, "Mtendere"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Afolayan"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "08/12/1994"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "05/12/1972"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "37"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Gbangadu MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Gvozden Boskovsky"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0"
}, "Inyene"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Okonkwo"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "04/01/1971"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "03/16/2000"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "70"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Kunike Barina"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Oscar de la Cavaller\xEDa"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0"
}, "Amaka"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Pretorius"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/25/1996"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "09/15/1986"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "32"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Bargbo"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Alberto Raya"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0"
}, "Meti"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Abiodun"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "10/24/2010"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/26/1989"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "8"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Majihun MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Unassigned"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0"
}, "Eshe"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Okeke"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/31/1995"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/31/1995"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "63"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Mambiama CHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Shadrias Pearson"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0"
}, "Obi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Okafor"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "06/07/1990"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/03/2006"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "28"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Dalakuru CHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Anatoliy Shcherbatykh"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete"))));

const FixedFirstColumn = FixedFirstColumnTemplate.bind({});
exports.FixedFirstColumn = FixedFirstColumn;
FixedFirstColumn.args = {
  scrollWidth: '500px'
};

const FixedHeaderAndTwoColumnsTemplate = args => /*#__PURE__*/_react.default.createElement(_dataTable.DataTable, args, /*#__PURE__*/_react.default.createElement(_dataTableHead.DataTableHead, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0",
  left: "0",
  width: "120px"
}, "First name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0",
  left: "120px"
}, "Last name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Incident date"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Last updated"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Age"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Registering unit"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Assigned user"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Status"))), /*#__PURE__*/_react.default.createElement(_dataTableBody.DataTableBody, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Onyekachukwu"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Kariuki"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "05/25/1972"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "66"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Jawi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Sofie Hubert"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Kwasi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okafor"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "08/11/2010"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/26/1991"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "38"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Mokassie MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Dashonte Clarke"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Siyabonga"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Abiodun"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/21/1981"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "98"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Bathurst MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Unassigned"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Chiyembekezo"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okeke"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/23/1982"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/15/2003"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "2"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Mayolla MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Wan Gengxin"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Mtendere"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Afolayan"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "08/12/1994"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "05/12/1972"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "37"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Gbangadu MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Gvozden Boskovsky"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Inyene"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okonkwo"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "04/01/1971"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "03/16/2000"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "70"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Kunike Barina"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Oscar de la Cavaller\xEDa"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Amaka"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Pretorius"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/25/1996"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "09/15/1986"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "32"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Bargbo"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Alberto Raya"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Meti"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Abiodun"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "10/24/2010"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/26/1989"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "8"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Majihun MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Unassigned"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Eshe"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okeke"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/31/1995"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/31/1995"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "63"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Mambiama CHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Shadrias Pearson"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Obi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okafor"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "06/07/1990"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/03/2006"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "28"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Dalakuru CHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Anatoliy Shcherbatykh"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete"))));

const FixedHeaderAndTwoColumns = FixedHeaderAndTwoColumnsTemplate.bind({});
exports.FixedHeaderAndTwoColumns = FixedHeaderAndTwoColumns;
FixedHeaderAndTwoColumns.args = {
  layout: 'fixed',
  width: '1000px',
  scrollWidth: '500px',
  scrollHeight: '400px'
};

const ScrollingDataTableWithToolbarsTemplate = args => /*#__PURE__*/_react.default.createElement(_uiCore.Box, {
  width: "500px"
}, /*#__PURE__*/_react.default.createElement(_dataTableToolbar.DataTableToolbar, null, /*#__PURE__*/_react.default.createElement("p", null, "Content")), /*#__PURE__*/_react.default.createElement(_dataTable.DataTable, _extends({}, args, {
  layout: "fixed",
  width: "1000px",
  scrollWidth: "500px",
  scrollHeight: "400px"
}), /*#__PURE__*/_react.default.createElement(_dataTableHead.DataTableHead, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0",
  left: "0",
  width: "120px"
}, "First name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0",
  left: "120px"
}, "Last name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Incident date"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Last updated"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Age"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Registering unit"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Assigned user"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
  fixed: true,
  top: "0"
}, "Status"))), /*#__PURE__*/_react.default.createElement(_dataTableBody.DataTableBody, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Onyekachukwu"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Kariuki"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "05/25/1972"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "66"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Jawi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Sofie Hubert"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Kwasi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okafor"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "08/11/2010"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/26/1991"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "38"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Mokassie MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Dashonte Clarke"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Siyabonga"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Abiodun"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/21/1981"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "98"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Bathurst MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Unassigned"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Chiyembekezo"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okeke"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/23/1982"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/15/2003"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "2"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Mayolla MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Wan Gengxin"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Mtendere"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Afolayan"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "08/12/1994"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "05/12/1972"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "37"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Gbangadu MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Gvozden Boskovsky"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Inyene"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okonkwo"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "04/01/1971"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "03/16/2000"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "70"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Kunike Barina"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Oscar de la Cavaller\xEDa"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Amaka"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Pretorius"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/25/1996"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "09/15/1986"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "32"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Bargbo"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Alberto Raya"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Meti"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Abiodun"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "10/24/2010"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "07/26/1989"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "8"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Majihun MCHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Unassigned"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Eshe"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okeke"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/31/1995"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/31/1995"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "63"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Mambiama CHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Shadrias Pearson"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Obi"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, {
  fixed: true,
  left: "120px"
}, "Okafor"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "06/07/1990"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "01/03/2006"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "28"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Dalakuru CHP"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Anatoliy Shcherbatykh"), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, "Incomplete")))), /*#__PURE__*/_react.default.createElement(_dataTableToolbar.DataTableToolbar, {
  position: "bottom"
}, /*#__PURE__*/_react.default.createElement("p", null, "Content (bottom)")));

const ScrollingDataTableWithToolbars = ScrollingDataTableWithToolbarsTemplate.bind({});
exports.ScrollingDataTableWithToolbars = ScrollingDataTableWithToolbars;
ScrollingDataTableWithToolbars.args = {
  layout: 'fixed',
  width: '1000px',
  scrollWidth: '500px',
  scrollHeight: '400px'
};

const ColumnHeaderSortingTemplate = args => {
  const rows = [{
    firstName: 'Onyekachukwu',
    lastName: 'Kariuki'
  }, {
    firstName: 'Kwasi',
    lastName: 'Okafor'
  }, {
    firstName: 'Siyabonga',
    lastName: 'Abiodun'
  }];
  const [{
    column,
    direction
  }, setSortInstructions] = (0, _react.useState)({
    column: 'firstName',
    direction: 'default'
  });

  const getSortDirection = columnName => columnName === column ? direction : 'default';

  const onSortIconClick = ({
    name,
    direction
  }) => {
    setSortInstructions({
      column: name,
      direction
    });
  };

  return /*#__PURE__*/_react.default.createElement(_dataTable.DataTable, args, /*#__PURE__*/_react.default.createElement(_dataTableHead.DataTableHead, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
    onSortIconClick: onSortIconClick,
    sortDirection: getSortDirection('firstName'),
    name: 'firstName'
  }, "First name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
    onSortIconClick: onSortIconClick,
    sortDirection: getSortDirection('lastName'),
    name: 'lastName'
  }, "Last name"))), /*#__PURE__*/_react.default.createElement(_dataTableBody.DataTableBody, null, rows.sort((a, b) => {
    const strA = a[column];
    const strB = b[column];

    if (direction === 'asc' && strA < strB || direction === 'desc' && strA > strB) {
      return -1;
    }

    if (direction === 'desc' && strA < strB || direction === 'asc' && strA > strB) {
      return 1;
    }

    return 0;
  }).map(({
    firstName,
    lastName
  }) => /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, {
    key: firstName + lastName
  }, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, firstName), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, lastName)))));
};

const ColumnHeaderSorting = ColumnHeaderSortingTemplate.bind({});
exports.ColumnHeaderSorting = ColumnHeaderSorting;
ColumnHeaderSorting.args = {};

const InlineFilteringTemplate = args => {
  const rows = [{
    firstName: 'Onyekachukwu',
    lastName: 'Kariuki'
  }, {
    firstName: 'Kwasi',
    lastName: 'Okafor'
  }, {
    firstName: 'Siyabonga',
    lastName: 'Abiodun'
  }];
  const [{
    column,
    value
  }, setFilter] = (0, _react.useState)({
    column: null,
    value: ''
  });

  const onFilterIconClick = ({
    name,
    show
  }) => {
    setFilter({
      column: show ? name : null,
      value: ''
    });
  };

  const onFilterInputChange = ({
    value
  }) => {
    setFilter({
      column: column,
      value
    });
  };

  return /*#__PURE__*/_react.default.createElement(_dataTable.DataTable, args, /*#__PURE__*/_react.default.createElement(_dataTableHead.DataTableHead, null, /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, null, /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
    onFilterIconClick: onFilterIconClick,
    name: "firstName",
    showFilter: column === 'firstName',
    filter: /*#__PURE__*/_react.default.createElement(_uiCore.Input, {
      dense: true,
      onChange: onFilterInputChange,
      name: "firstName",
      value: value
    })
  }, "First name"), /*#__PURE__*/_react.default.createElement(_dataTableColumnHeader.DataTableColumnHeader, {
    onFilterIconClick: onFilterIconClick,
    name: "lastName",
    showFilter: column === 'lastName',
    filter: /*#__PURE__*/_react.default.createElement(_uiCore.Input, {
      dense: true,
      onChange: onFilterInputChange,
      name: "lastName",
      value: value
    })
  }, "Last name"))), /*#__PURE__*/_react.default.createElement(_dataTableBody.DataTableBody, null, rows.filter(row => {
    if (!column || !value) {
      return true;
    }

    return row[column].toUpperCase().includes(value.toUpperCase());
  }).map(({
    firstName,
    lastName
  }) => /*#__PURE__*/_react.default.createElement(_dataTableRow.DataTableRow, {
    key: firstName + lastName
  }, /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, firstName), /*#__PURE__*/_react.default.createElement(_dataTableCell.DataTableCell, null, lastName)))));
};

const InlineFiltering = InlineFilteringTemplate.bind({});
exports.InlineFiltering = InlineFiltering;
InlineFiltering.args = {
  layout: 'fixed'
};
