// @dhis2-ui/button/build/cjs/button-strip/button-strip.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ButtonStrip = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _propTypes = require("@dhis2/prop-types");

var _uiConstants = require("@dhis2/ui-constants");

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes2 = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const ButtonStrip = ({
  className,
  children,
  middle,
  end,
  dataTest
}) => /*#__PURE__*/_react.default.createElement("div", {
  "data-test": dataTest,
  className: _style.default.dynamic([["2823325675", [_uiConstants.spacers.dp16]]]) + " " + ((0, _classnames.default)(className, {
    start: !middle && !end,
    middle,
    end
  }) || "")
}, _react.Children.map(children, child => /*#__PURE__*/_react.default.createElement("div", {
  className: _style.default.dynamic([["2823325675", [_uiConstants.spacers.dp16]]]) + " " + "box"
}, child)), /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "2823325675",
  dynamic: [_uiConstants.spacers.dp16]
}, ["div.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}", "div.middle.__jsx-style-dynamic-selector{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}", "div.end.__jsx-style-dynamic-selector{-webkit-box-pack:end;-webkit-justify-content:flex-end;-ms-flex-pack:end;justify-content:flex-end;}", `.box.__jsx-style-dynamic-selector{margin-left:${_uiConstants.spacers.dp16};}`, ".box.__jsx-style-dynamic-selector:first-child{margin-left:0;}"]));

exports.ButtonStrip = ButtonStrip;
const alignmentPropType = (0, _propTypes.mutuallyExclusive)(['middle', 'end'], _propTypes2.default.bool);
ButtonStrip.defaultProps = {
  dataTest: 'dhis2-uicore-buttonstrip'
};
ButtonStrip.propTypes = {
  children: _propTypes2.default.node,
  className: _propTypes2.default.string,
  dataTest: _propTypes2.default.string,

  /** Horizontal alignment for buttons. Mutually exclusive with `middle` prop */
  end: alignmentPropType,

  /** Horizontal alignment. Mutually exclusive with `end` prop */
  middle: alignmentPropType
};
