// @dhis2-ui/button/build/cjs/button-strip/button-strip.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DefaultAlignedRight = exports.DefaultAlignedMiddle = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _index = require("../index.js");

var _index2 = require("./index.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const description = `
A wrapper for buttons to add spacing and alignment.

\`\`\`js
import { ButtonStrip } from '@dhis2/ui'
\`\`\`
`;

const Wrapper = fn => /*#__PURE__*/_react.default.createElement("div", {
  style: {
    width: '100%',
    border: '1px solid #c4c9cc',
    padding: 8
  }
}, fn());

const alignmentArgType = {
  table: {
    type: {
      summary: 'bool',
      detail: "'middle' and 'end' are mutually exclusive props"
    }
  },
  control: {
    type: 'boolean'
  }
};
var _default = {
  title: 'Actions/Buttons/Button Strip',
  component: _index2.ButtonStrip,
  decorators: [Wrapper],
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  },
  argTypes: {
    middle: { ...alignmentArgType
    },
    end: { ...alignmentArgType
    }
  }
};
exports.default = _default;

const Default = args => /*#__PURE__*/_react.default.createElement(_index2.ButtonStrip, args, /*#__PURE__*/_react.default.createElement(_index.Button, null, "Save"), /*#__PURE__*/_react.default.createElement(_index.Button, null, "Save"), /*#__PURE__*/_react.default.createElement(_index.Button, null, "Save"), /*#__PURE__*/_react.default.createElement(_index.Button, null, "Save"), /*#__PURE__*/_react.default.createElement(_index.SplitButton, null, "Label?"));

exports.Default = Default;

const DefaultAlignedMiddle = args => /*#__PURE__*/_react.default.createElement(_index2.ButtonStrip, args, /*#__PURE__*/_react.default.createElement(_index.Button, null, "Save"), /*#__PURE__*/_react.default.createElement(_index.Button, null, "Save"), /*#__PURE__*/_react.default.createElement(_index.Button, null, "Save"), /*#__PURE__*/_react.default.createElement(_index.Button, null, "Save"));

exports.DefaultAlignedMiddle = DefaultAlignedMiddle;
DefaultAlignedMiddle.args = {
  middle: true
};
DefaultAlignedMiddle.storyName = 'Default - aligned middle';

const DefaultAlignedRight = args => /*#__PURE__*/_react.default.createElement(_index2.ButtonStrip, args, /*#__PURE__*/_react.default.createElement(_index.Button, null, "Save"), /*#__PURE__*/_react.default.createElement(_index.Button, null, "Save"), /*#__PURE__*/_react.default.createElement(_index.Button, null, "Save"), /*#__PURE__*/_react.default.createElement(_index.Button, null, "Save"));

exports.DefaultAlignedRight = DefaultAlignedRight;
DefaultAlignedRight.args = {
  end: true
};
DefaultAlignedRight.storyName = 'Default - aligned right';
