// @dhis2-ui/css/build/cjs/css-variables/css-variables.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NoVariables = exports.AllVariables = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _index = require("./index.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const description = `
A utility for adding DHIS2 theme variables to global CSS variables.

\`\`\`js
import { CssVariables } from '@dhis2/ui'
\`\`\`
`;

const App = ({
  children
}) => /*#__PURE__*/_react.default.createElement("div", null, children);

var _default = {
  title: 'Helpers/CSS Variables',
  component: _index.CssVariables,
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  }
};
exports.default = _default;

const AllVariables = args => /*#__PURE__*/_react.default.createElement(App, null, /*#__PURE__*/_react.default.createElement(_index.CssVariables, args), /*#__PURE__*/_react.default.createElement("p", null, "The sections of the theme that should be inserted can be toggled with flags, which allows the theme variables to be used as regular CSS custom properties. So this", ' ', /*#__PURE__*/_react.default.createElement("span", {
  style: {
    color: 'var(--colors-red500)'
  }
}, "text"), " uses the vanilla CSS", ' ', /*#__PURE__*/_react.default.createElement("span", {
  style: {
    color: 'var(--colors-blue500)'
  }
}, "custom properties"), ' ', "set by the CssVariables component."));

exports.AllVariables = AllVariables;
AllVariables.args = {
  colors: true,
  theme: true,
  layers: true,
  spacers: true,
  elevations: true
};

const NoVariables = args => /*#__PURE__*/_react.default.createElement(App, null, /*#__PURE__*/_react.default.createElement(_index.CssVariables, args), /*#__PURE__*/_react.default.createElement("p", null, "By default no custom properties are inserted."));

exports.NoVariables = NoVariables;
