// @dhis2-ui/label/build/es/label.js
import _JSXStyle from "styled-jsx/style";
import { Required } from '@dhis2-ui/required';
import propTypes from '@dhis2/prop-types';
import cx from 'classnames';
import React from 'react';
const styles = ["label.jsx-2718078600{display:block;box-sizing:border-box;font-size:14px;line-height:24px;padding:0;}", ".disabled.jsx-2718078600{cursor:not-allowed;}"];
styles.__hash = "2718078600";

const constructClassName = ({
  disabled,
  className
}) => cx(className, {
  disabled: disabled
});

export const Label = ({
  htmlFor,
  children,
  required,
  disabled,
  className,
  dataTest
}) => /*#__PURE__*/React.createElement("label", {
  htmlFor: htmlFor,
  "data-test": dataTest,
  className: `jsx-${styles.__hash}` + " " + (constructClassName({
    className,
    disabled
  }) || "")
}, /*#__PURE__*/React.createElement("span", {
  className: `jsx-${styles.__hash}`
}, children), required && /*#__PURE__*/React.createElement(Required, {
  dataTest: `${dataTest}-required`
}), /*#__PURE__*/React.createElement(_JSXStyle, {
  id: styles.__hash
}, styles));
Label.defaultProps = {
  dataTest: 'dhis2-uicore-label'
};
Label.propTypes = {
  children: propTypes.string,
  className: propTypes.string,
  dataTest: propTypes.string,
  disabled: propTypes.bool,
  htmlFor: propTypes.string,
  required: propTypes.bool
};
