// @dhis2-ui/radio/build/cjs/radio.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NoLabel = exports.ImageLabelDense = exports.ErrorDense = exports.WarningDense = exports.ValidDense = exports.DisabledDense = exports.CheckedDense = exports.FocusedCheckedDense = exports.FocusedUncheckedDense = exports.DefaultDense = exports.ImageLabel = exports.Error = exports.Warning = exports.Valid = exports.Disabled = exports.Checked = exports.FocusedChecked = exports.FocusedUnchecked = exports.Default = exports.default = void 0;

var _uiConstants = require("@dhis2/ui-constants");

var _react = _interopRequireDefault(require("react"));

var _radio = require("./radio.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const subtitle = `A control that allows a user to select a single option from a choice of several`;
const description = `
Radio buttons are used where a user has the choice of several options but must select only one. Radio buttons should be used where the user has to make a choice, there is no 'off' or 'none' state unless explicitly defined. Radio buttons should be used when there are 5 or less options available. With more than five, a dropdown/Select menu should be used instead.

Do not use a radio button if only a single option is available; use a Checkbox here instead.

If there are many options that need to select from, consider using a Select instead.

#### Size

Radio buttons are available in Regular and Dense sizes. Regular size is usually used in forms and whenever radio buttons are used standalone. Dense size radio buttons are used inside other complex components, not as main elements of a UI.

#### See more

Learn more about Radio buttons at [Design System: Radio](https://github.com/dhis2/design-system/blob/master/atoms/radio.md).

\`\`\`js
import { Radio } from '@dhis2/ui'
\`\`\`
`;

window.onChange = (payload, event) => {
  console.log('onChange payload', payload);
  console.log('onChange event', event);
};

window.onFocus = (payload, event) => {
  console.log('onFocus payload', payload);
  console.log('onFocus event', event);
};

window.onBlur = (payload, event) => {
  console.log('onBlur payload', payload);
  console.log('onBlur event', event);
};

const onChange = (...args) => window.onChange(...args);

const onFocus = (...args) => window.onFocus(...args);

const onBlur = (...args) => window.onBlur(...args);

var _default = {
  title: 'Forms/Radio/Radio',
  component: _radio.Radio,
  parameters: {
    componentSubtitle: subtitle,
    docs: {
      description: {
        component: description
      }
    }
  },
  // Default args for all stories
  args: {
    name: 'Ex',
    label: 'Radio',
    value: 'default',
    onChange: onChange,
    onFocus: onFocus,
    onBlur: onBlur
  },
  argTypes: {
    valid: { ..._uiConstants.sharedPropTypes.statusArgType
    },
    error: { ..._uiConstants.sharedPropTypes.statusArgType
    },
    warning: { ..._uiConstants.sharedPropTypes.statusArgType
    }
  }
};
exports.default = _default;

const Template = args => /*#__PURE__*/_react.default.createElement(_radio.Radio, args);

const CheckedUncheckedTemplate = args => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_radio.Radio, args), /*#__PURE__*/_react.default.createElement(_radio.Radio, _extends({}, args, {
  checked: true
})));

const Default = Template.bind({});
exports.Default = Default;

const FocusedUnchecked = args => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_radio.Radio, _extends({}, args, {
  initialFocus: true,
  className: "initially-focused"
})), /*#__PURE__*/_react.default.createElement(_radio.Radio, _extends({}, args, {
  className: "initially-unfocused"
}))); // Stories with initial focus are distracting on docs page


exports.FocusedUnchecked = FocusedUnchecked;
FocusedUnchecked.parameters = {
  docs: {
    disable: true
  }
};
const FocusedChecked = FocusedUnchecked.bind({});
exports.FocusedChecked = FocusedChecked;
FocusedChecked.args = {
  checked: true
};
FocusedChecked.parameters = {
  docs: {
    disable: true
  }
};
const Checked = Template.bind({});
exports.Checked = Checked;
Checked.args = {
  checked: true,
  value: 'checked'
};
const Disabled = CheckedUncheckedTemplate.bind({});
exports.Disabled = Disabled;
Disabled.args = {
  disabled: true,
  value: 'disabled'
};
const Valid = CheckedUncheckedTemplate.bind({});
exports.Valid = Valid;
Valid.args = {
  valid: true,
  value: 'valid'
};
const Warning = CheckedUncheckedTemplate.bind({});
exports.Warning = Warning;
Warning.args = {
  warning: true,
  value: 'warning'
};
const Error = CheckedUncheckedTemplate.bind({});
exports.Error = Error;
Error.args = {
  error: true,
  value: 'error'
};
const ImageLabel = Template.bind({});
exports.ImageLabel = ImageLabel;
ImageLabel.args = {
  label: /*#__PURE__*/_react.default.createElement("img", {
    src: "https://picsum.photos/id/82/200/100"
  }),
  value: 'with-help'
};
const DefaultDense = Template.bind({});
exports.DefaultDense = DefaultDense;
DefaultDense.args = {
  dense: true
};
DefaultDense.storyName = 'Default - Dense';
const FocusedUncheckedDense = FocusedUnchecked.bind({});
exports.FocusedUncheckedDense = FocusedUncheckedDense;
FocusedUncheckedDense.args = { ...DefaultDense.args
};
FocusedUncheckedDense.storyName = 'Focused unchecked - Dense';
FocusedUncheckedDense.parameters = {
  docs: {
    disable: true
  }
};
const FocusedCheckedDense = FocusedUnchecked.bind({});
exports.FocusedCheckedDense = FocusedCheckedDense;
FocusedCheckedDense.args = { ...DefaultDense.args,
  checked: true
};
FocusedCheckedDense.storyName = 'Focused checked - Dense';
FocusedCheckedDense.parameters = {
  docs: {
    disable: true
  }
};
const CheckedDense = Template.bind({});
exports.CheckedDense = CheckedDense;
CheckedDense.args = { ...Checked.args,
  ...DefaultDense.args
};
CheckedDense.storyName = 'Checked - Dense';
const DisabledDense = CheckedUncheckedTemplate.bind({});
exports.DisabledDense = DisabledDense;
DisabledDense.args = { ...Disabled.args,
  ...DefaultDense.args
};
DisabledDense.storyName = 'Disabled - Dense';
const ValidDense = CheckedUncheckedTemplate.bind({});
exports.ValidDense = ValidDense;
ValidDense.args = { ...Valid.args,
  ...DefaultDense.args
};
ValidDense.storyName = 'Valid - Dense';
const WarningDense = CheckedUncheckedTemplate.bind({});
exports.WarningDense = WarningDense;
WarningDense.args = { ...Warning.args,
  ...DefaultDense.args
};
WarningDense.storyName = 'Warning - Dense';
const ErrorDense = CheckedUncheckedTemplate.bind({});
exports.ErrorDense = ErrorDense;
ErrorDense.args = { ...Error.args,
  ...DefaultDense.args
};
ErrorDense.storyName = 'Error - Dense';
const ImageLabelDense = Template.bind({});
exports.ImageLabelDense = ImageLabelDense;
ImageLabelDense.args = { ...ImageLabel.args,
  ...DefaultDense.args
};
ImageLabelDense.storyName = 'Image label - Dense';
const NoLabel = Template.bind({});
exports.NoLabel = NoLabel;
NoLabel.args = {
  label: null,
  className: 'some-name'
};
