// @dhis2-ui/table/build/es/table/table-cell.js
import _JSXStyle from "styled-jsx/style";
import cx from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
const tableCellStyles = ["td.jsx-996330601{border-bottom:1px solid #e8edf2;font-size:14px;line-height:18px;padding:13px 12px;height:45px;}", ".dense.jsx-996330601{padding:9px 12px;height:36px;}"];
tableCellStyles.__hash = "996330601";
export const TableCell = ({
  role,
  className,
  children,
  colSpan,
  rowSpan,
  dense,
  dataTest
}) => /*#__PURE__*/React.createElement("td", {
  colSpan: colSpan,
  rowSpan: rowSpan,
  "data-test": dataTest,
  role: role,
  className: `jsx-${tableCellStyles.__hash}` + " " + (cx({
    dense
  }, className) || "")
}, children, /*#__PURE__*/React.createElement(_JSXStyle, {
  id: tableCellStyles.__hash
}, tableCellStyles));
TableCell.defaultProps = {
  dataTest: 'dhis2-uicore-tablecell'
};
TableCell.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  colSpan: PropTypes.string,
  dataTest: PropTypes.string,

  /** Usees less padding and height for information-dense layouts */
  dense: PropTypes.bool,
  role: PropTypes.string,
  rowSpan: PropTypes.string
};
