// @dhis2-ui/file-input/build/es/file-input-field-with-list/file-list-item-with-remove.js
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import propTypes from '@dhis2/prop-types';
import React, { Component } from 'react';
import { FileListItem } from '../index.js';

class FileListItemWithRemove extends Component {
  constructor(...args) {
    super(...args);

    _defineProperty(this, "handleRemove", event => {
      const {
        onRemove,
        file
      } = this.props;
      onRemove({
        file
      }, event);
    });
  }

  render() {
    const {
      label,
      removeText,
      className
    } = this.props;
    return /*#__PURE__*/React.createElement(FileListItem, {
      label: label,
      removeText: removeText,
      onRemove: this.handleRemove,
      className: className
    });
  }

}

FileListItemWithRemove.propTypes = {
  label: propTypes.string.isRequired,
  removeText: propTypes.string.isRequired,
  onRemove: propTypes.func.isRequired,
  className: propTypes.string,
  file: propTypes.instanceOf(File)
};
export { FileListItemWithRemove };
