// @dhis2-ui/divider/build/cjs/divider.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Margin = exports.Dense = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _divider = require("./divider.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const description = `
A light divider to separate content.

\`\`\`js
import { Divider } from '@dhis2/ui'
\`\`\`
`;
var _default = {
  title: 'Layout/Divider',
  component: _divider.Divider,
  argTypes: {
    margin: {
      table: {
        defaultValue: {
          summary: '8px 0'
        }
      }
    }
  },
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  }
};
exports.default = _default;

const Template = args => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", null, "Content above"), /*#__PURE__*/_react.default.createElement(_divider.Divider, args), /*#__PURE__*/_react.default.createElement("div", null, "Content below"));

const Default = Template.bind({});
exports.Default = Default;
const Dense = Template.bind({});
exports.Dense = Dense;
Dense.args = {
  dense: true
};
const Margin = Template.bind({});
exports.Margin = Margin;
Margin.args = {
  margin: '20px 20px 20px 20px'
};
