// @dhis2-ui/divider/build/cjs/divider.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Divider = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _uiConstants = require("@dhis2/ui-constants");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const Divider = ({
  className,
  dataTest,
  dense,
  margin
}) => /*#__PURE__*/_react.default.createElement("div", {
  "data-test": dataTest,
  className: _style.default.dynamic([["3889267139", [_uiConstants.colors.grey300]], ["3486645161", [dense ? `${_uiConstants.spacers.dp4} 0` : margin]]]) + " " + (className || "")
}, /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "3889267139",
  dynamic: [_uiConstants.colors.grey300]
}, [`div.__jsx-style-dynamic-selector{display:inline-block;width:100%;height:1px;background-color:${_uiConstants.colors.grey300};}`]), /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "3486645161",
  dynamic: [dense ? `${_uiConstants.spacers.dp4} 0` : margin]
}, [`div.__jsx-style-dynamic-selector{margin:${dense ? `${_uiConstants.spacers.dp4} 0` : margin};}`]));

exports.Divider = Divider;
Divider.defaultProps = {
  dataTest: 'dhis2-uicore-divider',
  margin: `${_uiConstants.spacers.dp8} 0`
};
Divider.propTypes = {
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string,
  dense: _propTypes.default.bool,
  margin: _propTypes.default.string
};
