// @dhis2-ui/select/build/cjs/select/empty.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Empty = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _propTypes = _interopRequireDefault(require("@dhis2/prop-types"));

var _uiConstants = require("@dhis2/ui-constants");

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const Empty = ({
  message,
  className,
  dataTest
}) => /*#__PURE__*/_react.default.createElement("div", {
  "data-test": dataTest,
  className: _style.default.dynamic([["849993029", [_uiConstants.colors.grey700, _uiConstants.theme.fonts, _uiConstants.spacers.dp8, _uiConstants.spacers.dp24]]]) + " " + (className || "")
}, message, /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "849993029",
  dynamic: [_uiConstants.colors.grey700, _uiConstants.theme.fonts, _uiConstants.spacers.dp8, _uiConstants.spacers.dp24]
}, [`div.__jsx-style-dynamic-selector{color:${_uiConstants.colors.grey700};font-family:${_uiConstants.theme.fonts};font-size:13px;line-height:16px;padding:${_uiConstants.spacers.dp8} ${_uiConstants.spacers.dp24};text-align:center;}`]));

exports.Empty = Empty;
Empty.propTypes = {
  dataTest: _propTypes.default.string.isRequired,
  message: _propTypes.default.string.isRequired,
  className: _propTypes.default.string
};
