// @dhis2-ui/select/build/cjs/select/filter-input.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FilterInput = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _input = require("@dhis2-ui/input");

var _propTypes = _interopRequireDefault(require("@dhis2/prop-types"));

var _uiConstants = require("@dhis2/ui-constants");

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const FilterInput = ({
  value,
  onChange,
  placeholder,
  className,
  dataTest
}) => /*#__PURE__*/_react.default.createElement("div", {
  className: _style.default.dynamic([["1544004171", [_uiConstants.colors.white, _uiConstants.spacers.dp8, _uiConstants.spacers.dp8, _uiConstants.spacers.dp4, _uiConstants.spacers.dp8]]]) + " " + (className || "")
}, /*#__PURE__*/_react.default.createElement(_input.Input, {
  dense: true,
  dataTest: dataTest,
  value: value,
  onChange: onChange,
  type: "text",
  name: "filter",
  placeholder: placeholder,
  initialFocus: true
}), /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "1544004171",
  dynamic: [_uiConstants.colors.white, _uiConstants.spacers.dp8, _uiConstants.spacers.dp8, _uiConstants.spacers.dp4, _uiConstants.spacers.dp8]
}, [`div.__jsx-style-dynamic-selector{position:-webkit-sticky;position:sticky;top:0;background:${_uiConstants.colors.white};padding:${_uiConstants.spacers.dp8} ${_uiConstants.spacers.dp8} ${_uiConstants.spacers.dp4} ${_uiConstants.spacers.dp8};z-index:1;}`]));

exports.FilterInput = FilterInput;
FilterInput.propTypes = {
  dataTest: _propTypes.default.string.isRequired,
  value: _propTypes.default.string.isRequired,
  className: _propTypes.default.string,
  placeholder: _propTypes.default.string,
  onChange: _propTypes.default.func
};
