// @dhis2-ui/select/build/cjs/select/input-prefix.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InputPrefix = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _propTypes = _interopRequireDefault(require("@dhis2/prop-types"));

var _uiConstants = require("@dhis2/ui-constants");

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const InputPrefix = ({
  prefix,
  className,
  dataTest
}) => {
  if (!prefix) {
    return null;
  }

  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": dataTest,
    className: _style.default.dynamic([["153011991", [_uiConstants.colors.grey600, _uiConstants.spacers.dp4]]]) + " " + (className || "")
  }, prefix, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "153011991",
    dynamic: [_uiConstants.colors.grey600, _uiConstants.spacers.dp4]
  }, [`div.__jsx-style-dynamic-selector{color:${_uiConstants.colors.grey600};padding-right:${_uiConstants.spacers.dp4};font-size:14px;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;}`]));
};

exports.InputPrefix = InputPrefix;
InputPrefix.propTypes = {
  dataTest: _propTypes.default.string.isRequired,
  className: _propTypes.default.string,
  prefix: _propTypes.default.string
};
