// @dhis2-ui/select/build/cjs/select/menu-wrapper.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MenuWrapper = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _card = require("@dhis2-ui/card");

var _layer = require("@dhis2-ui/layer");

var _popper = require("@dhis2-ui/popper");

var _propTypes = _interopRequireDefault(require("@dhis2/prop-types"));

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const MenuWrapper = ({
  children,
  dataTest,
  maxHeight,
  menuWidth,
  onClick,
  selectRef
}) => {
  const {
    styles,
    className: cardClassName
  } = {
    styles: /*#__PURE__*/_react.default.createElement(_style.default, {
      id: "81273804",
      dynamic: [maxHeight]
    }, [`.__jsx-style-dynamic-selector{height:auto;max-height:${maxHeight};overflow:auto;}`]),
    className: _style.default.dynamic([["81273804", [maxHeight]]])
  };
  return /*#__PURE__*/_react.default.createElement(_layer.Layer, {
    onClick: onClick,
    transparent: true
  }, /*#__PURE__*/_react.default.createElement(_popper.Popper, {
    reference: selectRef,
    placement: "bottom-start",
    observeReferenceResize: true
  }, /*#__PURE__*/_react.default.createElement("div", {
    "data-test": `${dataTest}-menuwrapper`,
    className: _style.default.dynamic([["3936600096", [menuWidth]]])
  }, /*#__PURE__*/_react.default.createElement(_card.Card, {
    className: cardClassName
  }, children), styles, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3936600096",
    dynamic: [menuWidth]
  }, [`div.__jsx-style-dynamic-selector{width:${menuWidth};}`]))));
};

exports.MenuWrapper = MenuWrapper;
MenuWrapper.defaultProps = {
  maxHeight: '280px'
};
MenuWrapper.propTypes = {
  dataTest: _propTypes.default.string.isRequired,
  menuWidth: _propTypes.default.string.isRequired,
  selectRef: _propTypes.default.object.isRequired,
  children: _propTypes.default.node,
  maxHeight: _propTypes.default.string,
  onClick: _propTypes.default.func
};
