// @dhis2-ui/alert/build/cjs/alert-stack/alert-stack.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _index = require("../alert-bar/index.js");

var _alertStack = require("./alert-stack.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const description = `
A container for Alert Bars.

_**Note:** The demos on this page may be slow._

\`\`\`js
import { AlertStack } from '@dhis2/ui'
\`\`\`
`;
var _default = {
  title: 'Feedback/Alerts/Alert Stack',
  component: _alertStack.AlertStack,
  // Use an iframe in docs to contain 'portal'
  parameters: {
    docs: {
      inlineStories: false,
      iframeHeight: '300px',
      description: {
        component: description
      }
    }
  }
};
exports.default = _default;

const Default = args => /*#__PURE__*/_react.default.createElement(_alertStack.AlertStack, args, /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
  permanent: true
}, "First notification - I am at the bottom"), /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
  permanent: true,
  critical: true
}, "Second notification"), /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
  permanent: true,
  warning: true
}, "Third notification"), /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
  permanent: true,
  success: true
}, "Fourth notification - I am at the top"));

exports.Default = Default;
