// @dhis2-ui/select/build/es/select/loading.js
import _JSXStyle from "styled-jsx/style";
import { CircularLoader } from '@dhis2-ui/loader';
import propTypes from '@dhis2/prop-types';
import { colors, spacers, theme } from '@dhis2/ui-constants';
import React from 'react';
export const Loading = ({
  message,
  className,
  dataTest
}) => /*#__PURE__*/React.createElement("div", {
  "data-test": dataTest,
  className: _JSXStyle.dynamic([["1434277130", [colors.grey700, theme.fonts, spacers.dp8, spacers.dp24]]]) + " " + (className || "")
}, /*#__PURE__*/React.createElement(CircularLoader, {
  small: true
}), message, /*#__PURE__*/React.createElement(_JSXStyle, {
  id: "1434277130",
  dynamic: [colors.grey700, theme.fonts, spacers.dp8, spacers.dp24]
}, [`div.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;color:${colors.grey700};font-family:${theme.fonts};font-size:13px;padding:${spacers.dp8} ${spacers.dp24};}`]));
Loading.propTypes = {
  dataTest: propTypes.string.isRequired,
  className: propTypes.string,
  message: propTypes.string
};
