// @dhis2-ui/select/build/es/select/select.js
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import propTypes from '@dhis2/prop-types';
import { sharedPropTypes } from '@dhis2/ui-constants';
import React, { Component } from 'react';
import { debounce } from './debounce/debounce.js';
import { InputWrapper } from './input-wrapper.js';
import { MenuWrapper } from './menu-wrapper.js'; // Keycodes for the keypress event handlers

const ESCAPE_KEY = 27;
const SPACE_KEY = 32;
const UP_KEY = 38;
const DOWN_KEY = 40;
export class Select extends Component {
  constructor(...args) {
    super(...args);

    _defineProperty(this, "state", {
      open: false,
      menuWidth: 'auto'
    });

    _defineProperty(this, "selectRef", /*#__PURE__*/React.createRef());

    _defineProperty(this, "inputRef", /*#__PURE__*/React.createRef());

    _defineProperty(this, "onResize", debounce(() => {
      const menuWidth = this.getMenuWidth();

      if (this.state.menuWidth !== menuWidth) {
        this.setState({
          menuWidth
        });
      }
    }, 50));

    _defineProperty(this, "handleFocusInput", () => {
      this.inputRef.current.focus();
    });

    _defineProperty(this, "onFocus", e => {
      const {
        onFocus,
        disabled,
        selected
      } = this.props;

      if (disabled || !onFocus) {
        return;
      }

      onFocus({
        selected
      }, e);
    });

    _defineProperty(this, "handleOpen", () => {
      this.setState({
        open: true,
        menuWidth: this.getMenuWidth()
      });
    });

    _defineProperty(this, "handleClose", () => {
      this.setState({
        open: false
      });
    });

    _defineProperty(this, "onToggle", e => {
      if (this.props.disabled) {
        return;
      }

      e.stopPropagation();
      this.state.open ? this.handleClose() : this.handleOpen();
    });

    _defineProperty(this, "onOutsideClick", e => {
      const {
        onBlur,
        disabled,
        selected
      } = this.props;

      if (disabled) {
        return;
      }

      this.handleClose();

      if (onBlur) {
        onBlur({
          selected
        }, e);
      }
    });

    _defineProperty(this, "onKeyDown", e => {
      if (this.props.disabled) {
        return;
      }

      e.stopPropagation();
      const {
        open
      } = this.state;
      const {
        keyCode
      } = e;
      const shouldOpen = !open && (keyCode === SPACE_KEY || keyCode === UP_KEY || keyCode === DOWN_KEY);
      const shouldClose = open && keyCode === ESCAPE_KEY;

      if (shouldClose) {
        return this.handleClose();
      }

      if (shouldOpen) {
        return this.handleOpen();
      }
    });
  }

  componentDidMount() {
    if (this.props.initialFocus) {
      this.inputRef.current.focus();
    }

    this.setState({
      menuWidth: this.getMenuWidth()
    });
    window.addEventListener('resize', this.onResize);
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.onResize);
  }
  /**
   * We're debouncing this so it doesn't fire continually during a resize.
   *
   * Additionally we should use requestPostAnimationFrame to not trigger a forced
   * layout, but that's just a proposal, and the added complexity of solving this
   * in another manner does not seem worth it, considering the minor perf penalty.
   *
   * See: https://nolanlawson.com/2018/09/25/accurately-measuring-layout-on-the-web
   */


  getMenuWidth() {
    const {
      offsetWidth
    } = this.inputRef.current;
    const {
      menuWidth
    } = this.state;

    if (offsetWidth && `${offsetWidth}px` !== menuWidth) {
      return `${offsetWidth}px`;
    }

    return menuWidth;
  }

  render() {
    const {
      open,
      menuWidth
    } = this.state;
    const {
      children,
      className,
      selected,
      onChange,
      tabIndex,
      maxHeight,
      error,
      warning,
      valid,
      disabled,
      dense,
      dataTest
    } = this.props; // Create the input

    const inputProps = {
      selected,
      onChange,
      options: children,
      disabled
    };
    const input = /*#__PURE__*/React.cloneElement(this.props.input, inputProps); // Create the menu

    const menuProps = {
      selected,
      onChange,
      options: children,
      handleClose: this.handleClose,
      handleFocusInput: this.handleFocusInput
    };
    const menu = /*#__PURE__*/React.cloneElement(this.props.menu, menuProps);
    return /*#__PURE__*/React.createElement("div", {
      className: className,
      ref: this.selectRef,
      onFocus: this.onFocus,
      onKeyDown: this.onKeyDown,
      "data-test": dataTest
    }, /*#__PURE__*/React.createElement(InputWrapper, {
      onToggle: this.onToggle,
      inputRef: this.inputRef,
      tabIndex: tabIndex,
      error: error,
      warning: warning,
      valid: valid,
      disabled: disabled,
      dense: dense,
      dataTest: `${dataTest}-input`
    }, input), open && /*#__PURE__*/React.createElement(MenuWrapper, {
      onClick: this.onOutsideClick,
      maxHeight: maxHeight,
      selectRef: this.selectRef,
      menuWidth: menuWidth,
      dataTest: `${dataTest}-menu`
    }, menu));
  }

}
Select.defaultProps = {
  dataTest: 'dhis2-uicore-select'
};
Select.propTypes = {
  input: propTypes.element.isRequired,
  menu: propTypes.element.isRequired,
  selected: propTypes.oneOfType([propTypes.string, propTypes.arrayOf(propTypes.string)]).isRequired,
  children: propTypes.node,
  className: propTypes.string,
  dataTest: propTypes.string,
  dense: propTypes.bool,
  disabled: propTypes.bool,
  error: sharedPropTypes.statusPropType,
  initialFocus: propTypes.bool,
  maxHeight: propTypes.string,
  tabIndex: propTypes.string,
  valid: sharedPropTypes.statusPropType,
  warning: sharedPropTypes.statusPropType,
  onBlur: propTypes.func,
  onChange: propTypes.func,
  onFocus: propTypes.func
};
