// @dhis2-ui/alert/build/es/alert-stack/alert-stack.js
import _JSXStyle from "styled-jsx/style";
import { Portal } from '@dhis2-ui/portal';
import { layers } from '@dhis2/ui-constants';
import cx from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
export const AlertStack = ({
  className,
  children,
  dataTest
}) => /*#__PURE__*/React.createElement(Portal, null, /*#__PURE__*/React.createElement("div", {
  "data-test": dataTest,
  className: _JSXStyle.dynamic([["132471376", [layers.alert]]]) + " " + (cx(className) || "")
}, children, /*#__PURE__*/React.createElement(_JSXStyle, {
  id: "132471376",
  dynamic: [layers.alert]
}, [`div.__jsx-style-dynamic-selector{position:fixed;top:auto;right:auto;bottom:0;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);z-index:${layers.alert};display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column-reverse;-ms-flex-direction:column-reverse;flex-direction:column-reverse;pointer-events:none;}`])));
AlertStack.defaultProps = {
  dataTest: 'dhis2-uicore-alertstack'
};
AlertStack.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  dataTest: PropTypes.string
};
