// @dhis2-ui/alert/build/es/alert-stack/alert-stack.stories.js
import React from 'react';
import { AlertBar } from '../alert-bar/index.js';
import { AlertStack } from './alert-stack.js';
const description = `
A container for Alert Bars.

_**Note:** The demos on this page may be slow._

\`\`\`js
import { AlertStack } from '@dhis2/ui'
\`\`\`
`;
export default {
  title: 'Feedback/Alerts/Alert Stack',
  component: AlertStack,
  // Use an iframe in docs to contain 'portal'
  parameters: {
    docs: {
      inlineStories: false,
      iframeHeight: '300px',
      description: {
        component: description
      }
    }
  }
};
export const Default = args => /*#__PURE__*/React.createElement(AlertStack, args, /*#__PURE__*/React.createElement(AlertBar, {
  permanent: true
}, "First notification - I am at the bottom"), /*#__PURE__*/React.createElement(AlertBar, {
  permanent: true,
  critical: true
}, "Second notification"), /*#__PURE__*/React.createElement(AlertBar, {
  permanent: true,
  warning: true
}, "Third notification"), /*#__PURE__*/React.createElement(AlertBar, {
  permanent: true,
  success: true
}, "Fourth notification - I am at the top"));
