// @dhis2-ui/select/build/cjs/single-select/single-select.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ShiftedIntoView = exports.FlippedPosition = exports.DefaultPosition = exports.WithFilterField = exports.WithClearButtonSelectionAndOnChange = exports.WithDisabledOptionAndOnChange = exports.WithPlaceholderAndSelection = exports.WithPlaceholder = exports.WithPrefixAndSelection = exports.WithPrefix = exports.WithOptionsASelectionAndDisabled = exports.WithOptionsAndDisabled = exports.WithMoreThanThreeOptionsAndA100PxMaxHeight = exports.WithMoreThanTenOptions = exports.WithOptionsLoadingAndLoadingText = exports.WithOptionsAndLoading = exports.EmptyWithEmptyComponent = exports.EmptyWithEmptyText = exports.Empty = exports.WithInitialFocus = exports.WithInvalidFilterableOptions = exports.WithInvalidOptions = exports.WithCustomOptionsAndOnChange = exports.WithOnBlur = exports.WithOnFocus = exports.WithOptionsAndASelection = exports.WithOptionsAndOnChange = exports.default = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _propTypes = _interopRequireDefault(require("@dhis2/prop-types"));

var _uiConstants = require("@dhis2/ui-constants");

var _react = _interopRequireDefault(require("react"));

var _index = require("../index.js");

var _index2 = require("./index.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const description = `
Use a select component wherever the user needs to make a selection of one or more options from a list of 6 or more options. If there are less than 6 options to choose from and space permits, use checkboxes for multiple selection and radio buttons for single selection. If the user needs to make a complex selection with a specific ordering, use a transfer instead.

Learn more about using Selects at [Design System: Select](https://github.com/dhis2/design-system/blob/master/molecules/select.md).

\`\`\`js
import { SingleSelect, SingleSelectOption } from '@dhis2/ui'
\`\`\`

_**Note:** Due to demo limitations on this page, only one representative example is rendered here. For more (interactive) examples, see individual stories in the 'Canvas' tab._
`;

const eventHandler = handlerName => (payload, event) => {
  console.log(`${handlerName} payload`, payload);
  console.log(`${handlerName} event`, event);
};

const onChange = eventHandler('onChange');
const onFocus = eventHandler('onFocus');
const onBlur = eventHandler('onBlur');

const CustomSingleSelectOption = ({
  label,
  onClick
}) => /*#__PURE__*/_react.default.createElement("div", {
  onClick: e => onClick({}, e)
}, label);

CustomSingleSelectOption.propTypes = {
  label: _propTypes.default.string,
  onClick: _propTypes.default.func
};
const requiredIfArgType = {
  table: {
    type: {
      summary: 'string'
    },
    control: {
      type: 'string'
    }
  }
};
var _default = {
  title: 'Forms/Single Select/Single Select',
  component: _index2.SingleSelect,
  parameters: {
    docs: {
      description: {
        component: description
      },
      // Disable stories in docs page by default, then use one representative examples
      disable: true,
      // Use iframe to keep dropdown menu inside story for docs page
      inlineStories: false,
      iframeHeight: '300px'
    }
  },
  // Use 'onChange' as a default arg, otherwise components throw an error when interacted with
  // (Maybe this could be handled in the component - as a required prop)
  args: {
    className: 'select',
    onChange
  },
  argTypes: {
    valid: { ..._uiConstants.sharedPropTypes.statusArgType
    },
    warning: { ..._uiConstants.sharedPropTypes.statusArgType
    },
    error: { ..._uiConstants.sharedPropTypes.statusArgType
    },
    clearText: { ...requiredIfArgType
    },
    noMatchText: { ...requiredIfArgType
    }
  }
};
exports.default = _default;

const WithOptionsTemplate = args => /*#__PURE__*/_react.default.createElement(_index2.SingleSelect, args, /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "3",
  label: "option three"
}));

const EmptyTemplate = args => /*#__PURE__*/_react.default.createElement(_index2.SingleSelect, args);

const WithOptionsAndOnChange = WithOptionsTemplate.bind({});
exports.WithOptionsAndOnChange = WithOptionsAndOnChange;
WithOptionsAndOnChange.storyName = 'With options and onChange';
const WithOptionsAndASelection = WithOptionsTemplate.bind({});
exports.WithOptionsAndASelection = WithOptionsAndASelection;
WithOptionsAndASelection.args = {
  selected: '1'
}; // Enable this story as the primary for docs as a representative example

WithOptionsAndASelection.parameters = {
  docs: {
    disable: false
  }
};
WithOptionsAndASelection.storyName = 'With options and a selection';
const WithOnFocus = WithOptionsTemplate.bind({});
exports.WithOnFocus = WithOnFocus;
WithOnFocus.args = {
  onFocus
};
WithOnFocus.storyName = 'With onFocus';
const WithOnBlur = WithOptionsTemplate.bind({});
exports.WithOnBlur = WithOnBlur;
WithOnBlur.args = {
  onBlur
};
WithOnBlur.storyName = 'With onBlur';

const WithCustomOptionsAndOnChange = args => /*#__PURE__*/_react.default.createElement(_index2.SingleSelect, args, /*#__PURE__*/_react.default.createElement(CustomSingleSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/_react.default.createElement(CustomSingleSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/_react.default.createElement(CustomSingleSelectOption, {
  value: "3",
  label: "option three"
}));

exports.WithCustomOptionsAndOnChange = WithCustomOptionsAndOnChange;
WithCustomOptionsAndOnChange.storyName = 'With custom options and onChange';

const WithInvalidOptions = args => /*#__PURE__*/_react.default.createElement(_index2.SingleSelect, args, /*#__PURE__*/_react.default.createElement("div", null, "invalid one"), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/_react.default.createElement("div", null, "invalid two"), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/_react.default.createElement("div", null, "invalid three"), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "3",
  label: "option three"
}), null, undefined, false);

exports.WithInvalidOptions = WithInvalidOptions;

const WithInvalidFilterableOptions = args => /*#__PURE__*/_react.default.createElement(_index2.SingleSelect, args, /*#__PURE__*/_react.default.createElement("div", null, "invalid one"), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/_react.default.createElement("div", null, "invalid two"), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/_react.default.createElement("div", null, "invalid three"), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "3",
  label: "option three"
}));

exports.WithInvalidFilterableOptions = WithInvalidFilterableOptions;
WithInvalidFilterableOptions.args = {
  filterable: true,
  noMatchText: 'No match found'
};
const WithInitialFocus = EmptyTemplate.bind({});
exports.WithInitialFocus = WithInitialFocus;
WithInitialFocus.args = {
  initialFocus: true
};
WithInitialFocus.parameters = {
  docs: {
    disable: true
  }
};
WithInitialFocus.storyName = 'With initialFocus';
const Empty = EmptyTemplate.bind({});
exports.Empty = Empty;
const EmptyWithEmptyText = EmptyTemplate.bind({});
exports.EmptyWithEmptyText = EmptyWithEmptyText;
EmptyWithEmptyText.args = {
  empty: 'Custom empty text'
};
const EmptyWithEmptyComponent = EmptyTemplate.bind({});
exports.EmptyWithEmptyComponent = EmptyWithEmptyComponent;
EmptyWithEmptyComponent.args = {
  empty: /*#__PURE__*/_react.default.createElement("div", {
    className: "custom-empty"
  }, "Custom empty component")
};
const WithOptionsAndLoading = WithOptionsTemplate.bind({});
exports.WithOptionsAndLoading = WithOptionsAndLoading;
WithOptionsAndLoading.args = {
  loading: true
};
const WithOptionsLoadingAndLoadingText = WithOptionsTemplate.bind({});
exports.WithOptionsLoadingAndLoadingText = WithOptionsLoadingAndLoadingText;
WithOptionsLoadingAndLoadingText.args = { ...WithOptionsAndLoading.args,
  loadingText: 'Loading options'
};
WithOptionsLoadingAndLoadingText.storyName = 'With options, loading and loading text';

const WithMoreThanTenOptions = args => /*#__PURE__*/_react.default.createElement(_index2.SingleSelect, args, /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "3",
  label: "option three"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "4",
  label: "option four"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "5",
  label: "option five"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "6",
  label: "option six"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "7",
  label: "option seven"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "8",
  label: "option eight"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "9",
  label: "option nine"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "10",
  label: "option ten"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "11",
  label: "option eleven"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "12",
  label: "option twelve"
}));

exports.WithMoreThanTenOptions = WithMoreThanTenOptions;

const WithMoreThanThreeOptionsAndA100PxMaxHeight = args => /*#__PURE__*/_react.default.createElement(_index2.SingleSelect, args, /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "3",
  label: "option three"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "4",
  label: "option four"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "5",
  label: "option five"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "6",
  label: "option six"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "7",
  label: "option seven"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "8",
  label: "option eight"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "9",
  label: "option nine"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "10",
  label: "option ten"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "11",
  label: "option eleven"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "12",
  label: "option twelve"
}));

exports.WithMoreThanThreeOptionsAndA100PxMaxHeight = WithMoreThanThreeOptionsAndA100PxMaxHeight;
WithMoreThanThreeOptionsAndA100PxMaxHeight.args = {
  maxHeight: '100px'
};
WithMoreThanThreeOptionsAndA100PxMaxHeight.storyName = 'With more than three options and a 100px max-height';
const WithOptionsAndDisabled = WithOptionsTemplate.bind({});
exports.WithOptionsAndDisabled = WithOptionsAndDisabled;
WithOptionsAndDisabled.args = {
  disabled: true
};
WithOptionsAndDisabled.storyName = 'With options and disabled';
const WithOptionsASelectionAndDisabled = WithOptionsTemplate.bind({});
exports.WithOptionsASelectionAndDisabled = WithOptionsASelectionAndDisabled;
WithOptionsASelectionAndDisabled.args = { ...WithOptionsAndDisabled.args,
  ...WithOptionsAndASelection.args
};
WithOptionsASelectionAndDisabled.storyName = 'With options, a selection and disabled';
const WithPrefix = WithOptionsTemplate.bind({});
exports.WithPrefix = WithPrefix;
WithPrefix.args = {
  prefix: 'Prefix text'
};
const WithPrefixAndSelection = WithOptionsTemplate.bind({});
exports.WithPrefixAndSelection = WithPrefixAndSelection;
WithPrefixAndSelection.args = { ...WithPrefix.args,
  ...WithOptionsAndASelection.args
};
const WithPlaceholder = WithOptionsTemplate.bind({});
exports.WithPlaceholder = WithPlaceholder;
WithPlaceholder.args = {
  placeholder: 'Placeholder text'
};
const WithPlaceholderAndSelection = WithOptionsTemplate.bind({});
exports.WithPlaceholderAndSelection = WithPlaceholderAndSelection;
WithPlaceholderAndSelection.args = { ...WithPlaceholder.args,
  ...WithOptionsAndASelection.args
};

const WithDisabledOptionAndOnChange = args => /*#__PURE__*/_react.default.createElement(_index2.SingleSelect, args, /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "3",
  label: "option three"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "4",
  label: "disabled option",
  disabled: true
}));

exports.WithDisabledOptionAndOnChange = WithDisabledOptionAndOnChange;
WithDisabledOptionAndOnChange.storyName = 'With disabled option and onChange';
const WithClearButtonSelectionAndOnChange = WithOptionsTemplate.bind({});
exports.WithClearButtonSelectionAndOnChange = WithClearButtonSelectionAndOnChange;
WithClearButtonSelectionAndOnChange.args = { ...WithOptionsAndASelection.args,
  clearable: true,
  clearText: 'Clear'
};
WithClearButtonSelectionAndOnChange.storyName = 'With clear button, selection and onChange';
const WithFilterField = WithOptionsTemplate.bind({});
exports.WithFilterField = WithFilterField;
WithFilterField.args = { ...WithInvalidFilterableOptions.args
};

const DefaultPosition = args => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_index2.SingleSelect, args, /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "3",
  label: "option three"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "4",
  label: "option four"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "5",
  label: "option five"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "6",
  label: "option six"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "7",
  label: "option seven"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "8",
  label: "option eight"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "9",
  label: "option nine"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "10",
  label: "option ten"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "11",
  label: "option eleven"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "12",
  label: "option twelve"
})), /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "2006483503"
}, ["#root{margin-bottom:2000px;}"]));

exports.DefaultPosition = DefaultPosition;

const FlippedPosition = args => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_index2.SingleSelect, args, /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "3",
  label: "option three"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "4",
  label: "option four"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "5",
  label: "option five"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "6",
  label: "option six"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "7",
  label: "option seven"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "8",
  label: "option eight"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "9",
  label: "option nine"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "10",
  label: "option ten"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "11",
  label: "option eleven"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "12",
  label: "option twelve"
})), /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "1520050670"
}, ["html,body,#root{position:relative;height:500px;max-height:500px;}", "#root{padding-top:400px !important;}"]));

exports.FlippedPosition = FlippedPosition;

const ShiftedIntoView = args => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_index2.SingleSelect, args, /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "1",
  label: "option one"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "2",
  label: "option two"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "3",
  label: "option three"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "4",
  label: "option four"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "5",
  label: "option five"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "6",
  label: "option six"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "7",
  label: "option seven"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "8",
  label: "option eight"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "9",
  label: "option nine"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "10",
  label: "option ten"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "11",
  label: "option eleven"
}), /*#__PURE__*/_react.default.createElement(_index.SingleSelectOption, {
  value: "12",
  label: "option twelve"
})), /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "1828862170"
}, ["html,body,#root{position:relative;height:300px !important;max-height:300px;}", "#root{padding-top:130px !important;}"]));

exports.ShiftedIntoView = ShiftedIntoView;
