// @dhis2-ui/button/build/cjs/dropdown-button/dropdown-button.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InitialFocus = exports.WithMenu = exports.Large = exports.Small = exports.Disabled = exports.Destructive = exports.Secondary = exports.Primary = exports.WithClick = exports.Default = exports.default = void 0;

var _menu = require("@dhis2-ui/menu");

var _uiConstants = require("@dhis2/ui-constants");

var _react = _interopRequireDefault(require("react"));

var _index = require("./index.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const description = `
Presents several actions to a user in a small space. Can replace single, individual buttons. Should only be used for actions that are related to one another. Ensure the button has a useful level that communicates that actions are contained within. Dropdown buttons do not have an explicit action, only expanding the list of contained actions.

\`\`\`js
import { DropdownButton } from '@dhis2/ui'
\`\`\`
`;

window.onClick = (payload, event) => {
  console.log('onClick payload', payload);
  console.log('onClick event', event);
};

const onClick = (...args) => window.onClick(...args);

const Simple = /*#__PURE__*/_react.default.createElement("span", null, "Simplest thing");

const {
  sizeArgType,
  buttonVariantArgType
} = _uiConstants.sharedPropTypes;
var _default = {
  title: 'Actions/Buttons/Dropdown Button',
  component: _index.DropdownButton,
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  },
  argTypes: {
    primary: { ...buttonVariantArgType
    },
    secondary: { ...buttonVariantArgType
    },
    destructive: { ...buttonVariantArgType
    },
    small: { ...sizeArgType
    },
    large: { ...sizeArgType
    }
  },
  // Default args for all stories (can be overridden)
  args: {
    name: 'buttonName',
    value: 'buttonValue',
    component: Simple,
    children: 'Label me!'
  }
};
exports.default = _default;

const Template = args => /*#__PURE__*/_react.default.createElement(_index.DropdownButton, args);

const Default = Template.bind({});
exports.Default = Default;
const WithClick = Template.bind({});
exports.WithClick = WithClick;
WithClick.args = {
  onClick: onClick
};
const Primary = Template.bind({});
exports.Primary = Primary;
Primary.args = {
  primary: true
};
const Secondary = Template.bind({});
exports.Secondary = Secondary;
Secondary.args = {
  secondary: true
};
const Destructive = Template.bind({});
exports.Destructive = Destructive;
Destructive.args = {
  destructive: true
};
const Disabled = Template.bind({});
exports.Disabled = Disabled;
Disabled.args = {
  disabled: true
};
const Small = Template.bind({});
exports.Small = Small;
Small.args = {
  small: true
};
const Large = Template.bind({});
exports.Large = Large;
Large.args = {
  large: true
};
const WithMenu = Template.bind({});
exports.WithMenu = WithMenu;
WithMenu.args = {
  component: /*#__PURE__*/_react.default.createElement(_menu.FlyoutMenu, null, /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
    label: "Item 1"
  }), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
    label: "Item 2"
  }), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
    label: "Item 3"
  }))
}; // FlyoutMenu needs iframe
// But docs page down too much with iframe, so disabled

WithMenu.parameters = {
  docs: {
    disable: true
  }
};
const InitialFocus = Template.bind({});
exports.InitialFocus = InitialFocus;
InitialFocus.args = {
  initialFocus: true
};
/**
 * 'Initial focus' stories cause the docs page to scroll away each time
 * a control is changed, therefore it is omitted from the docs page (but
 * not the normal 'canvas' story viewer)
 */

InitialFocus.parameters = {
  docs: {
    disable: true
  }
};
