// @dhis2-ui/help/build/es/help.js
import _JSXStyle from "styled-jsx/style";
import { spacers, theme, colors, sharedPropTypes } from '@dhis2/ui-constants';
import cx from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

const Help = ({
  children,
  valid,
  error,
  warning,
  className,
  dataTest
}) => /*#__PURE__*/React.createElement("p", {
  "data-test": dataTest,
  className: _JSXStyle.dynamic([["1707655029", [spacers.dp4, theme.default, colors.blue700, theme.error, colors.yellow800]]]) + " " + (cx(className, {
    valid,
    error,
    warning
  }) || "")
}, children, /*#__PURE__*/React.createElement(_JSXStyle, {
  id: "1707655029",
  dynamic: [spacers.dp4, theme.default, colors.blue700, theme.error, colors.yellow800]
}, [`p.__jsx-style-dynamic-selector{margin-top:${spacers.dp4};margin-right:0;margin-bottom:0;margin-left:0;font-size:12px;line-height:14px;color:${theme.default};cursor:help;}`, `.valid.__jsx-style-dynamic-selector{color:${colors.blue700};}`, `.error.__jsx-style-dynamic-selector{color:${theme.error};}`, `.warning.__jsx-style-dynamic-selector{color:${colors.yellow800};}`]));

Help.defaultProps = {
  dataTest: 'dhis2-uicore-help'
};
Help.propTypes = {
  children: PropTypes.string,
  className: PropTypes.string,
  dataTest: PropTypes.string,
  error: sharedPropTypes.statusPropType,
  valid: sharedPropTypes.statusPropType,
  warning: sharedPropTypes.statusPropType
};
export { Help };
