// @dhis2-ui/notice-box/build/cjs/notice-box-message.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NoticeBoxMessage = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _propTypes = _interopRequireDefault(require("@dhis2/prop-types"));

var _uiConstants = require("@dhis2/ui-constants");

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const NoticeBoxMessage = ({
  children,
  dataTest
}) => {
  if (!children) {
    return null;
  }

  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": dataTest,
    className: _style.default.dynamic([["3452101222", [_uiConstants.colors.grey900]]])
  }, children, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3452101222",
    dynamic: [_uiConstants.colors.grey900]
  }, [`div.__jsx-style-dynamic-selector{color:${_uiConstants.colors.grey900};font-size:14px;line-height:20px;}`]));
};

exports.NoticeBoxMessage = NoticeBoxMessage;
NoticeBoxMessage.propTypes = {
  dataTest: _propTypes.default.string.isRequired,
  children: _propTypes.default.node
};
