// @dhis2-ui/modal/build/cjs/modal-title/modal-title.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ModalTitle = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _propTypes = _interopRequireDefault(require("@dhis2/prop-types"));

var _uiConstants = require("@dhis2/ui-constants");

var _classnames = _interopRequireDefault(require("classnames"));

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const ModalTitle = ({
  children,
  dataTest
}) => /*#__PURE__*/_react.default.createElement("h1", {
  "data-test": dataTest,
  className: _style.default.dynamic([["1525509444", [_uiConstants.spacers.dp24, _uiConstants.spacers.dp24, _uiConstants.spacers.dp24]]]) + " " + ((0, _classnames.default)('title') || "")
}, children, /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "1525509444",
  dynamic: [_uiConstants.spacers.dp24, _uiConstants.spacers.dp24, _uiConstants.spacers.dp24]
}, [`h1.__jsx-style-dynamic-selector{font-size:20px;font-weight:500;line-height:24px;margin:0;padding:${_uiConstants.spacers.dp24} ${_uiConstants.spacers.dp24} 0 ${_uiConstants.spacers.dp24};}`]));

exports.ModalTitle = ModalTitle;
ModalTitle.defaultProps = {
  dataTest: 'dhis2-uicore-modaltitle'
};
ModalTitle.propTypes = {
  children: _propTypes.default.string,
  dataTest: _propTypes.default.string
};
