// @dhis2-ui/modal/build/cjs/modal/modal.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ModalThatHidesWithStatefulComponens = exports.LargeModalWithMoreNestedModals = exports.LargeWithSelectComponent = exports.SmallLongTitle = exports.BottomScrollable = exports.MiddleScrollable = exports.TopScrollable = exports.SmallClickableScreenCover = exports.SmallDestructivePrimary = exports.SmallContentPrimary = exports.LargeTitleContentPrimary = exports.MediumTitleContentAction = exports.SmallTitleContentAction = exports.AlignmentBottom = exports.AlignmentMiddle = exports.DefaultContent = exports.default = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _button = require("@dhis2-ui/button");

var _menu = require("@dhis2-ui/menu");

var _select = require("@dhis2-ui/select");

var _tooltip = require("@dhis2-ui/tooltip");

var _uiConstants = require("@dhis2/ui-constants");

var _react = _interopRequireWildcard(require("react"));

var _index = require("../index.js");

var _modal = require("./modal.js");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const description = `
_**Note**: For performance reasons, only one representative example is shown here. For more (interactive) examples, see individual stories in the Canvas tab._

A modal focuses user attention on a single task or piece of information inside a container. A modal blocks the rest of the application until it is dismissed.

#### Usage

A modal should be used to focus user attention on a single task or piece of information. Modals take over the entire screen and should be used sparingly to avoid interrupting a user's flow too often.

Use a modal in the following cases:

- to collect short, focussed user input that is blocking progress
- to present critical information that a user needs to acknowledge before continuing
- to ask the user to confirm a destructive action that cannot be undone

Do not use a modal in the following cases:

- to display unrelated or non-critical information; use an alert bar or a notice box instead
- to display complex, workflows that span multiple screens; navigate to a new, full page in the app instead

#### Children

Modals should be used with children \`<ModalTitle>\` (optional), \`<ModalContent>\` (required), and \`<ModalActions>\` (recommended)

See more about modal usage, including alignment and sizing, at [Design System: Modals](https://github.com/dhis2/design-system/blob/master/molecules/modal.md#usage).

\`\`\`js
import { Modal } from '@dhis2/ui'
\`\`\`
`;

const say = something => () => alert(something);

window.onClose = (payload, event) => {
  console.log('onClose payload', payload);
  console.log('onClose event', event);
};

const onClose = (...args) => window.onClose(...args);

var _default = {
  title: 'Layout/Modal',
  component: _modal.Modal,
  parameters: {
    docs: {
      // Use iframes to contain modals in docs page (otherwise chaos ensues)
      inlineStories: false,
      iframeHeight: '500px',

      /**
       * Due to iframes being very slow, disable stories on the docs page by default and
       * make one representative story as the primary ('SmallTitleContentAction')
       */
      disable: true,
      description: {
        component: description
      }
    }
  },
  argTypes: {
    small: { ..._uiConstants.sharedPropTypes.sizeArgType
    },
    large: { ..._uiConstants.sharedPropTypes.sizeArgType
    },
    position: { ..._uiConstants.sharedPropTypes.insideAlignmentArgType
    }
  }
};
exports.default = _default;

const DefaultContent = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."));

exports.DefaultContent = DefaultContent;
DefaultContent.args = {
  onClose: onClose
};
DefaultContent.storyName = 'Default: Content';

const AlignmentMiddle = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."));

exports.AlignmentMiddle = AlignmentMiddle;
AlignmentMiddle.args = {
  onClose,
  position: 'middle'
};
AlignmentMiddle.storyName = 'Alignment: Middle';

const AlignmentBottom = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."));

exports.AlignmentBottom = AlignmentBottom;
AlignmentBottom.args = {
  onClose,
  alignment: 'bottom'
};
AlignmentBottom.storyName = 'Alignment: Bottom';

const SmallTitleContentAction = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "This is a small modal with title, content and primary action"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action")))); // Have this be the primary story on the docs page


exports.SmallTitleContentAction = SmallTitleContentAction;
SmallTitleContentAction.parameters = {
  docs: {
    disable: false,
    source: {
      type: 'dynamic'
    }
  }
};
SmallTitleContentAction.args = {
  small: true,
  onClose
};
SmallTitleContentAction.storyName = 'Small: Title, Content, Action';

const MediumTitleContentAction = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "This is a medium modal with title, content and primary action"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action"))));

exports.MediumTitleContentAction = MediumTitleContentAction;
MediumTitleContentAction.storyName = 'Medium: Title, Content, Action';

const LargeTitleContentPrimary = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "This is a large modal with title, content and primary action"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action"))));

exports.LargeTitleContentPrimary = LargeTitleContentPrimary;
LargeTitleContentPrimary.args = {
  large: true
};
LargeTitleContentPrimary.storyName = 'Large: Title, Content, Primary';

const SmallContentPrimary = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action"))));

exports.SmallContentPrimary = SmallContentPrimary;
SmallContentPrimary.args = {
  small: true
};
SmallContentPrimary.storyName = 'Small: Content & Primary';

const SmallDestructivePrimary = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  destructive: true,
  onClick: say('Button primary')
}, "Primary action"))));

exports.SmallDestructivePrimary = SmallDestructivePrimary;
SmallDestructivePrimary.args = {
  small: true
};
SmallDestructivePrimary.storyName = 'Small: Destructive Primary';

const SmallClickableScreenCover = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "This is a modal with clickable screen cover"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua."), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  primary: true,
  onClick: say('Button primary')
}, "Primary action"))));

exports.SmallClickableScreenCover = SmallClickableScreenCover;
SmallClickableScreenCover.args = {
  small: true,
  onClose: args => {
    onClose(args);
    say('Clickable screen cover')();
  }
};
SmallClickableScreenCover.storyName = 'Small: Clickable screen cover';

const TopScrollable = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "This is a modal with scrollable content"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  destructive: true,
  onClick: say('Button primary')
}, "Primary action"))));

exports.TopScrollable = TopScrollable;
TopScrollable.args = { ...SmallClickableScreenCover.args
};
TopScrollable.storyName = 'Top: scrollable';

const MiddleScrollable = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "This is a modal with scrollable content"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  destructive: true,
  onClick: say('Button primary')
}, "Primary action"))));

exports.MiddleScrollable = MiddleScrollable;
MiddleScrollable.args = { ...TopScrollable.args,
  position: 'middle'
};
MiddleScrollable.storyName = 'Middle: scrollable';

const BottomScrollable = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "This is a modal with scrollable content"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  destructive: true,
  onClick: say('Button primary')
}, "Primary action"))));

exports.BottomScrollable = BottomScrollable;
BottomScrollable.args = { ...TopScrollable.args,
  position: 'bottom'
};
BottomScrollable.storyName = 'Bottom: scrollable';

const SmallLongTitle = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "This headline should break into multiple lines because it's way too long for one!"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum."), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action"))));

exports.SmallLongTitle = SmallLongTitle;
SmallLongTitle.args = {
  small: true
};
SmallLongTitle.storyName = 'Small: Long title';

const LargeWithSelectComponent = args => /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "Select opens on top of the Modal"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, /*#__PURE__*/_react.default.createElement(_select.SingleSelect, null, /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "1",
  value: "1",
  label: "one"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "2",
  value: "2",
  label: "two"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "3",
  value: "3",
  label: "three"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "4",
  value: "3",
  label: "four"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "5",
  value: "3",
  label: "five"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "6",
  value: "3",
  label: "six"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "7",
  value: "3",
  label: "seven"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "8",
  value: "3",
  label: "eight"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "9",
  value: "3",
  label: "nine"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "10",
  value: "3",
  label: "ten"
}))), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action"))));

exports.LargeWithSelectComponent = LargeWithSelectComponent;
LargeWithSelectComponent.args = {
  large: true
};
LargeWithSelectComponent.storyName = 'Large: with Select component';

const LargeModalWithMoreNestedModals = args => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "Select opens on top of the Modal - Level 1"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, /*#__PURE__*/_react.default.createElement(_select.SingleSelect, null, /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "1",
  value: "1",
  label: "one"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "2",
  value: "2",
  label: "two"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "3",
  value: "3",
  label: "three"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "4",
  value: "3",
  label: "four"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "5",
  value: "3",
  label: "five"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "6",
  value: "3",
  label: "six"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "7",
  value: "3",
  label: "seven"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "8",
  value: "3",
  label: "eight"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "9",
  value: "3",
  label: "nine"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "10",
  value: "3",
  label: "ten"
}))), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action")))), /*#__PURE__*/_react.default.createElement(_modal.Modal, args, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "Select opens on top of the Modal - Level 2"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, /*#__PURE__*/_react.default.createElement(_select.SingleSelect, null, /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "1",
  value: "1",
  label: "one"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "2",
  value: "2",
  label: "two"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "3",
  value: "3",
  label: "three"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "4",
  value: "3",
  label: "four"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "5",
  value: "3",
  label: "five"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "6",
  value: "3",
  label: "six"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "7",
  value: "3",
  label: "seven"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "8",
  value: "3",
  label: "eight"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "9",
  value: "3",
  label: "nine"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "10",
  value: "3",
  label: "ten"
}))), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action")))), /*#__PURE__*/_react.default.createElement(_modal.Modal, {
  small: true
}, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "Select opens on top of the Modal - Level 3"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, /*#__PURE__*/_react.default.createElement(_select.SingleSelect, null, /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "1",
  value: "1",
  label: "one"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "2",
  value: "2",
  label: "two"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "3",
  value: "3",
  label: "three"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "4",
  value: "3",
  label: "four"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "5",
  value: "3",
  label: "five"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "6",
  value: "3",
  label: "six"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "7",
  value: "3",
  label: "seven"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "8",
  value: "3",
  label: "eight"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "9",
  value: "3",
  label: "nine"
}), /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, {
  key: "10",
  value: "3",
  label: "ten"
})), /*#__PURE__*/_react.default.createElement(_tooltip.Tooltip, {
  content: "Some extra info"
}, ({
  ref,
  onMouseOver,
  onMouseOut
}) => /*#__PURE__*/_react.default.createElement("span", {
  ref: ref,
  onMouseOver: onMouseOver,
  onMouseOut: onMouseOut,
  className: "jsx-2364308943"
}, /*#__PURE__*/_react.default.createElement("button", {
  className: "jsx-2364308943"
}, "Tooltip on button"), /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "2364308943"
}, ["span.jsx-2364308943{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}"]))), /*#__PURE__*/_react.default.createElement(_menu.FlyoutMenu, null, /*#__PURE__*/_react.default.createElement(_menu.MenuSectionHeader, {
  label: "Section with sub-menus"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 1"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2"
}, /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2 a"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2 b"
}, /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2 b i"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2 b ii"
})), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2 c"
})), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 3"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 4"
}, /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 4 a"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 4 b"
}, /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 4 b i"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 4 b ii"
})), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 4 c"
})), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 5"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuSectionHeader, {
  label: "Section with dividers between menu items"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 1"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuDivider, null), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 2"
}), /*#__PURE__*/_react.default.createElement(_menu.MenuDivider, null), /*#__PURE__*/_react.default.createElement(_menu.MenuItem, {
  label: "Item 3"
}))), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
  end: true
}, /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button secondary'),
  secondary: true
}, "Secondary action"), /*#__PURE__*/_react.default.createElement(_button.Button, {
  onClick: say('Button primary'),
  primary: true
}, "Primary action")))));

exports.LargeModalWithMoreNestedModals = LargeModalWithMoreNestedModals;
LargeModalWithMoreNestedModals.args = {
  large: true
};
LargeModalWithMoreNestedModals.storyName = 'Large: modal with more nested modals';

const StatefuleComponent = () => {
  const [counter, setCounter] = (0, _react.useState)(0);
  return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("p", null, "Current counter: ", counter), /*#__PURE__*/_react.default.createElement("button", {
    onClick: () => setCounter(counter + 1)
  }, "Add 1 to counter"));
};

const ModalThatHidesWithStatefulComponens = () => {
  const [render, setRender] = (0, _react.useState)(false);
  const [hide, setHide] = (0, _react.useState)(false);
  return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, null, /*#__PURE__*/_react.default.createElement(_button.Button, {
    onClick: () => setHide(false),
    disabled: !render
  }, "Show the hidden modal"), /*#__PURE__*/_react.default.createElement(_button.Button, {
    onClick: () => setRender(true),
    disabled: render
  }, "Render Modal")), render && /*#__PURE__*/_react.default.createElement(_modal.Modal, {
    hide: hide
  }, /*#__PURE__*/_react.default.createElement(_index.ModalTitle, null, "Can be hidden"), /*#__PURE__*/_react.default.createElement(_index.ModalContent, null, /*#__PURE__*/_react.default.createElement(StatefuleComponent, null)), /*#__PURE__*/_react.default.createElement(_index.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
    end: true
  }, /*#__PURE__*/_react.default.createElement(_button.Button, {
    onClick: () => setRender(false),
    secondary: true
  }, "Close modal"), /*#__PURE__*/_react.default.createElement(_button.Button, {
    onClick: () => setHide(true),
    primary: true
  }, "Hide modal")))));
};

exports.ModalThatHidesWithStatefulComponens = ModalThatHidesWithStatefulComponens;
