// @dhis2-ui/table/build/cjs/table/table-cell-head.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TableCellHead = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const tableCellHeadStyles = ["th.jsx-1796049005{border-bottom:1px solid #e8edf2;font-size:14px;line-height:18px;padding:13px 12px;height:45px;}", ".dense.jsx-1796049005{padding:9px 12px;height:36px;}"];
tableCellHeadStyles.__hash = "1796049005";

const TableCellHead = ({
  role,
  colSpan,
  rowSpan,
  dense,
  children,
  className,
  dataTest
}) => /*#__PURE__*/_react.default.createElement("th", {
  colSpan: colSpan,
  rowSpan: rowSpan,
  "data-test": dataTest,
  role: role,
  className: `jsx-${tableCellHeadStyles.__hash}` + " " + ((0, _classnames.default)({
    dense
  }, className) || "")
}, children, /*#__PURE__*/_react.default.createElement(_style.default, {
  id: tableCellHeadStyles.__hash
}, tableCellHeadStyles));

exports.TableCellHead = TableCellHead;
TableCellHead.defaultProps = {
  dataTest: 'dhis2-uicore-tablecellhead'
};
TableCellHead.propTypes = {
  children: _propTypes.default.node,
  className: _propTypes.default.string,
  colSpan: _propTypes.default.string,
  dataTest: _propTypes.default.string,

  /** Uses less padding and height for information-dense layouts */
  dense: _propTypes.default.bool,
  role: _propTypes.default.string,
  rowSpan: _propTypes.default.string
};
