// @dhis2-ui/alert/build/es/alert-bar/dismiss.js
import _JSXStyle from "styled-jsx/style";
import propTypes from '@dhis2/prop-types';
import { spacers } from '@dhis2/ui-constants';
import { IconCross24 } from '@dhis2/ui-icons';
import React from 'react';

const Dismiss = ({
  onClick,
  dataTest
}) => /*#__PURE__*/React.createElement("div", {
  onClick: onClick,
  "data-test": dataTest,
  className: _JSXStyle.dynamic([["489362409", [spacers.dp16]]])
}, /*#__PURE__*/React.createElement(IconCross24, null), /*#__PURE__*/React.createElement(_JSXStyle, {
  id: "489362409",
  dynamic: [spacers.dp16]
}, [`div.__jsx-style-dynamic-selector{margin-left:${spacers.dp16};min-height:32px;min-width:32px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;border-radius:5px;}`, "div.__jsx-style-dynamic-selector:hover{cursor:pointer;background:rgba(0,0,0,0.15);}", "div.__jsx-style-dynamic-selector:active{background:rgba(0,0,0,0.25);}", "div.__jsx-style-dynamic-selector svg{width:18px;height:18px;}"]));

Dismiss.propTypes = {
  dataTest: propTypes.string.isRequired,
  onClick: propTypes.func.isRequired
};
export { Dismiss };
