// @dhis2-ui/radio/build/es/radio.js
import _JSXStyle from "styled-jsx/style";

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import { colors, theme, sharedPropTypes } from '@dhis2/ui-constants';
import cx from 'classnames';
import PropTypes from 'prop-types';
import React, { Component, createRef } from 'react';
import { RadioRegular, RadioDense } from './radio-icons.js';

class Radio extends Component {
  constructor(...args) {
    super(...args);

    _defineProperty(this, "ref", /*#__PURE__*/createRef());

    _defineProperty(this, "handleChange", e => {
      if (this.props.onChange) {
        this.props.onChange(this.createHandlerPayload(), e);
      }
    });

    _defineProperty(this, "handleBlur", e => {
      if (this.props.onBlur) {
        this.props.onBlur(this.createHandlerPayload(), e);
      }
    });

    _defineProperty(this, "handleFocus", e => {
      if (this.props.onFocus) {
        this.props.onFocus(this.createHandlerPayload(), e);
      }
    });
  }

  componentDidMount() {
    if (this.props.initialFocus) {
      this.ref.current.focus();
    }
  }

  createHandlerPayload() {
    return {
      value: this.props.value,
      name: this.props.name,
      checked: !this.props.checked
    };
  }

  render() {
    const {
      checked = false,
      className,
      disabled,
      error,
      label,
      name,
      tabIndex,
      valid,
      value,
      warning,
      dense,
      dataTest
    } = this.props;
    const classes = cx({
      checked,
      disabled,
      valid,
      error,
      warning
    });
    return /*#__PURE__*/React.createElement("label", {
      "data-test": dataTest,
      className: _JSXStyle.dynamic([["2233367747", [colors.grey900, theme.disabled, label ? '5px' : 0, theme.focus]]]) + " " + (cx(className, {
        disabled,
        dense
      }) || "")
    }, /*#__PURE__*/React.createElement("input", {
      type: "radio",
      ref: this.ref,
      name: name,
      value: value,
      checked: checked,
      disabled: disabled,
      tabIndex: tabIndex,
      onChange: this.handleChange,
      onFocus: this.handleFocus,
      onBlur: this.handleBlur,
      className: _JSXStyle.dynamic([["2233367747", [colors.grey900, theme.disabled, label ? '5px' : 0, theme.focus]]])
    }), /*#__PURE__*/React.createElement("div", {
      className: _JSXStyle.dynamic([["2233367747", [colors.grey900, theme.disabled, label ? '5px' : 0, theme.focus]]]) + " " + (cx('icon', {
        dense
      }) || "")
    }, dense ? /*#__PURE__*/React.createElement(RadioDense, {
      className: classes
    }) : /*#__PURE__*/React.createElement(RadioRegular, {
      className: classes
    })), label, /*#__PURE__*/React.createElement(_JSXStyle, {
      id: "2233367747",
      dynamic: [colors.grey900, theme.disabled, label ? '5px' : 0, theme.focus]
    }, [`label.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;position:relative;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;color:${colors.grey900};font-size:16px;line-height:20px;}`, "label.dense.__jsx-style-dynamic-selector{font-size:14px;line-height:16px;}", `label.disabled.__jsx-style-dynamic-selector{cursor:not-allowed;color:${theme.disabled};}`, "input.__jsx-style-dynamic-selector{opacity:0;position:absolute;height:18px;width:18px;left:3px;}", "label.dense.__jsx-style-dynamic-selector input.__jsx-style-dynamic-selector{height:14px;width:14px;}", `.icon.__jsx-style-dynamic-selector{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;margin-right:${label ? '5px' : 0};border:2px solid transparent;padding:1px;border-radius:50%;}`, "label.dense.__jsx-style-dynamic-selector .icon.__jsx-style-dynamic-selector{margin-right:3px;}", `input.__jsx-style-dynamic-selector:focus+.icon.__jsx-style-dynamic-selector{outline:3px solid ${theme.focus};outline-offset:-1px;}`]));
  }

}

Radio.defaultProps = {
  dataTest: 'dhis2-uicore-radio'
};
Radio.propTypes = {
  checked: PropTypes.bool,
  className: PropTypes.string,
  dataTest: PropTypes.string,
  dense: PropTypes.bool,
  disabled: PropTypes.bool,

  /** Adds 'error' styling for feedback. Mutually exclusive with `valid` and `warning` props */
  error: sharedPropTypes.statusPropType,
  initialFocus: PropTypes.bool,
  label: PropTypes.node,

  /** Name associated with this element. Passed in object to event handler functions  */
  name: PropTypes.string,
  tabIndex: PropTypes.string,

  /** Adds 'valid' styling for feedback. Mutually exclusive with `error` and `warning` props */
  valid: sharedPropTypes.statusPropType,

  /** Value associated with this element. Passed in object to event handler functions */
  value: PropTypes.string,

  /** Adds 'warning' styling for feedback. Mutually exclusive with `valid` and `error` props */
  warning: sharedPropTypes.statusPropType,

  /** Called with the signature `({ name: string, value: string, checked: bool }, event)` */
  onBlur: PropTypes.func,

  /** Called with the signature `({ name: string, value: string, checked: bool }, event)` */
  onChange: PropTypes.func,

  /** Called with the signature `({ name: string, value: string, checked: bool }, event)` */
  onFocus: PropTypes.func
};
export { Radio };
