// @dhis2-ui/table/build/cjs/data-table/table-elements/table.stories.internal.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.HeaderActions = exports.ScrollingTableWithToolbars = exports.FixedHeaderAndTwoColumns = exports.FixedFirstColumn = exports.FixedHeader = exports.SelectableRows = exports.Toolbars = exports.IrregularHeaders = exports.CellStyling = exports.DraggableRows = exports.StaticCells = exports.LargeCells = exports.BorderedCells = exports.Default = exports.default = void 0;

var _uiCore = require("@dhis2/ui-core");

var _uiIcons = require("@dhis2/ui-icons");

var _react = _interopRequireWildcard(require("react"));

var _tableBody = require("./table-body.js");

var _tableDataCell = require("./table-data-cell/table-data-cell.js");

var _tableFoot = require("./table-foot.js");

var _tableHead = require("./table-head.js");

var _tableHeaderCellAction = require("./table-header-cell-action.js");

var _tableHeaderCell = require("./table-header-cell/table-header-cell.js");

var _tableRow = require("./table-row.js");

var _tableScrollBox = require("./table-scroll-box.js");

var _tableToolbar = require("./table-toolbar.js");

var _table = require("./table.js");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

/* eslint-disable react/prop-types */
const subtitle = 'Used to display information in a standard, effective way.';
const description = `
Should be used with multiple Table-related child components - see the table and examples below.

\`\`\`js
import {
    Table,
    TableToolbar,
    TableHead,
    TableBody,
    TableFoot,
    TableRow,
    TableDataCell,
    TableHeaderCell,
    TableScrollBox,
} from '@dhis2/ui'
\`\`\`
`;
var _default = {
  title: 'Data Display/Table',
  component: _table.Table,
  // Add subcomponents to the args table
  subcomponents: {
    TableToolbar: _tableToolbar.TableToolbar,
    TableHead: _tableHead.TableHead,
    TableBody: _tableBody.TableBody,
    TableFoot: _tableFoot.TableFoot,
    TableRow: _tableRow.TableRow,
    TableDataCell: _tableDataCell.TableDataCell,
    TableHeaderCell: _tableHeaderCell.TableHeaderCell
  },
  parameters: {
    componentSubtitle: subtitle,
    docs: {
      description: {
        component: description
      }
    }
  },
  argTypes: {
    // hide control from `children` because that can't be live edited
    children: {
      control: {
        disable: true
      }
    },
    // hide child component controls completely to avoid confusion
    borderedCells: {
      table: {
        disable: true
      }
    },
    largeCells: {
      table: {
        disable: true
      }
    },
    staticCells: {
      table: {
        disable: true
      }
    },
    draggableRows: {
      table: {
        disable: true
      }
    }
  }
};
exports.default = _default;

const BasicTemplate = ({
  borderedCells,
  largeCells,
  staticCells,
  draggableRows,
  ...args
}) => /*#__PURE__*/_react.default.createElement(_table.Table, args, /*#__PURE__*/_react.default.createElement(_tableHead.TableHead, null, /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  large: largeCells
}, "First name"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  large: largeCells
}, "Last name"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  large: largeCells
}, "Incident date"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  large: largeCells
}, "Last updated"))), /*#__PURE__*/_react.default.createElement(_tableBody.TableBody, null, /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, {
  draggable: draggableRows
}, /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  staticStyle: staticCells,
  large: largeCells,
  bordered: borderedCells
}, "Onyekachukwu"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  staticStyle: staticCells,
  large: largeCells,
  bordered: borderedCells
}, "Kariuki"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  staticStyle: staticCells,
  large: largeCells,
  bordered: borderedCells
}, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  staticStyle: staticCells,
  large: largeCells,
  bordered: borderedCells
}, "05/25/1972")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, {
  draggable: draggableRows
}, /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  staticStyle: staticCells,
  large: largeCells,
  bordered: borderedCells
}, "Kwasi"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  staticStyle: staticCells,
  large: largeCells,
  bordered: borderedCells
}, "Okafor"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  staticStyle: staticCells,
  large: largeCells,
  bordered: borderedCells
}, "08/11/2010"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  staticStyle: staticCells,
  large: largeCells,
  bordered: borderedCells
}, "02/26/1991")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, {
  draggable: draggableRows
}, /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  staticStyle: staticCells,
  large: largeCells,
  bordered: borderedCells
}, "Siyabonga"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  staticStyle: staticCells,
  large: largeCells,
  bordered: borderedCells
}, "Abiodun"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  staticStyle: staticCells,
  large: largeCells,
  bordered: borderedCells
}, "07/21/1981"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  staticStyle: staticCells,
  large: largeCells,
  bordered: borderedCells
}, "02/06/2007"))), /*#__PURE__*/_react.default.createElement(_tableFoot.TableFoot, null, /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  staticStyle: staticCells,
  large: largeCells,
  bordered: borderedCells,
  colSpan: "4"
}, "Footer content"))));

const Default = BasicTemplate.bind({});
exports.Default = Default;
Default.args = {};
const BorderedCells = BasicTemplate.bind({});
exports.BorderedCells = BorderedCells;
BorderedCells.args = {
  borderedCells: true
};
const LargeCells = BasicTemplate.bind({});
exports.LargeCells = LargeCells;
LargeCells.args = {
  largeCells: true
};
const StaticCells = BasicTemplate.bind({});
exports.StaticCells = StaticCells;
StaticCells.args = {
  staticCells: true
};
const DraggableRows = BasicTemplate.bind({});
exports.DraggableRows = DraggableRows;
DraggableRows.args = {
  draggableRows: true
};

const IndividualCellTemplate = args => /*#__PURE__*/_react.default.createElement(_table.Table, args, /*#__PURE__*/_react.default.createElement(_tableHead.TableHead, null, /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, null, "Header"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, null, "Header"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, null, "Header"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, null, "Header"))), /*#__PURE__*/_react.default.createElement(_tableBody.TableBody, null, /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  tag: "th"
}, "Active Cells"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Inactive cell"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  active: true,
  onClick: () => {
    console.log('You clicked on a TableDataCell');
  }
}, "Acive Cell with onClick"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Inactive cell")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  tag: "th"
}, "Alignment"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  align: "left"
}, "Left"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  align: "center"
}, "Center"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  align: "right"
}, "Right")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  tag: "th"
}, "Built in formatting"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  error: true
}, "Error"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  valid: true
}, "Valid"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  muted: true
}, "Muted"))));

const CellStyling = IndividualCellTemplate.bind({});
exports.CellStyling = CellStyling;
CellStyling.args = {};

const IrregularHeadersTemplate = args => /*#__PURE__*/_react.default.createElement(_table.Table, args, /*#__PURE__*/_react.default.createElement("colgroup", null), /*#__PURE__*/_react.default.createElement("colgroup", {
  span: "2"
}), /*#__PURE__*/_react.default.createElement("colgroup", {
  span: "2"
}), /*#__PURE__*/_react.default.createElement(_tableHead.TableHead, null, /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  rowSpan: "2",
  colSpan: "2",
  bordered: true
}), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  align: "center",
  colSpan: "2",
  scope: "colgroup"
}, "Mars"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  align: "center",
  colSpan: "2",
  scope: "colgroup"
}, "Venus")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  scope: "col"
}, "Produced"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  scope: "col"
}, "Sold"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  scope: "col"
}, "Produced"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  scope: "col"
}, "Sold"))), /*#__PURE__*/_react.default.createElement(_tableBody.TableBody, null, /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  staticStyle: true,
  bordered: true,
  scope: "row",
  rowSpan: "2"
}, "Type"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  staticStyle: true,
  bordered: true,
  scope: "row"
}, "Teddy Bears"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  staticStyle: true
}, "50,000"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  staticStyle: true
}, "30,000"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  staticStyle: true
}, "100,000"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  staticStyle: true
}, "80,000")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  staticStyle: true,
  bordered: true,
  scope: "row"
}, "Board Games"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  staticStyle: true
}, "10,000"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  staticStyle: true
}, "5,000"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  staticStyle: true
}, "12,000"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  staticStyle: true
}, "9,000"))));

const IrregularHeaders = IrregularHeadersTemplate.bind({});
exports.IrregularHeaders = IrregularHeaders;
IrregularHeaders.args = {};

const ToolbarsTemplate = args => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_tableToolbar.TableToolbar, null, /*#__PURE__*/_react.default.createElement("p", null, "Content")), /*#__PURE__*/_react.default.createElement(_table.Table, args, /*#__PURE__*/_react.default.createElement(_tableHead.TableHead, null, /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, null, "First name"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, null, "Last name"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, null, "Incident date"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, null, "Last updated"))), /*#__PURE__*/_react.default.createElement(_tableBody.TableBody, null, /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Onyekachukwu"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Kariuki"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "05/25/1972")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Kwasi"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Okafor"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "08/11/2010"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "02/26/1991")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Siyabonga"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Abiodun"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "07/21/1981"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "02/06/2007")))), /*#__PURE__*/_react.default.createElement(_tableToolbar.TableToolbar, {
  position: "bottom"
}, /*#__PURE__*/_react.default.createElement("p", null, "Content (bottom)")));

const Toolbars = ToolbarsTemplate.bind({});
exports.Toolbars = Toolbars;
Toolbars.args = {};

const SelectableRowsTemplate = args => {
  const [selected, setSelected] = (0, _react.useState)({
    id_2: true
  });

  const toggleSelected = ({
    value,
    checked
  }) => {
    setSelected({ ...selected,
      [value]: checked
    });
  };

  const toggleAll = ({
    checked
  }) => {
    setSelected({
      id_1: checked,
      id_2: checked,
      id_3: checked
    });
  };

  const allSelected = () => Object.values(selected).filter(value => value).length === 3;

  return /*#__PURE__*/_react.default.createElement(_table.Table, args, /*#__PURE__*/_react.default.createElement(_tableHead.TableHead, null, /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
    width: "48px"
  }, /*#__PURE__*/_react.default.createElement(_uiCore.Checkbox, {
    checked: allSelected(),
    onChange: toggleAll
  })), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, null, "First name"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, null, "Last name"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, null, "Incident date"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, null, "Last updated"))), /*#__PURE__*/_react.default.createElement(_tableBody.TableBody, null, /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, {
    selected: !!selected.id_1
  }, /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
    width: "48px"
  }, /*#__PURE__*/_react.default.createElement(_uiCore.Checkbox, {
    value: "id_1",
    checked: !!selected.id_1,
    onChange: toggleSelected
  })), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Onyekachukwu"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Kariuki"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "05/25/1972")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, {
    selected: !!selected.id_2
  }, /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
    width: "48px"
  }, /*#__PURE__*/_react.default.createElement(_uiCore.Checkbox, {
    value: "id_2",
    checked: !!selected.id_2,
    onChange: toggleSelected
  })), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Kwasi"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Okafor"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "08/11/2010"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "02/26/1991")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, {
    selected: !!selected.id_3
  }, /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
    width: "48px"
  }, /*#__PURE__*/_react.default.createElement(_uiCore.Checkbox, {
    value: "id_3",
    checked: !!selected.id_3,
    onChange: toggleSelected
  })), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Siyabonga"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Abiodun"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "07/21/1981"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "02/06/2007"))));
};

const SelectableRows = SelectableRowsTemplate.bind({});
exports.SelectableRows = SelectableRows;
SelectableRows.args = {};

const FixedHeaderTemplate = ({ ...args
}) => /*#__PURE__*/_react.default.createElement(_tableScrollBox.TableScrollBox, {
  maxHeight: "350px"
}, /*#__PURE__*/_react.default.createElement(_table.Table, args, /*#__PURE__*/_react.default.createElement(_tableHead.TableHead, null, /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0"
}, "First name"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Last name"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Incident date"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Last updated"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Age"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Registering unit"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Assigned user"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Status"))), /*#__PURE__*/_react.default.createElement(_tableBody.TableBody, null, /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Onyekachukwu"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Kariuki"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "05/25/1972"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "66"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Jawi"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Sofie Hubert"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Kwasi"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Okafor"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "08/11/2010"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "02/26/1991"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "38"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Mokassie MCHP"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Dashonte Clarke"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Siyabonga"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Abiodun"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "07/21/1981"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "98"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Bathurst MCHP"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Unassigned"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Chiyembekezo"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Okeke"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "01/23/1982"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "07/15/2003"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "2"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Mayolla MCHP"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Wan Gengxin"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Mtendere"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Afolayan"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "08/12/1994"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "05/12/1972"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "37"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Gbangadu MCHP"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Gvozden Boskovsky"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Inyene"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Okonkwo"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "04/01/1971"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "03/16/2000"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "70"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Kunike Barina"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Oscar de la Cavaller\xEDa"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Amaka"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Pretorius"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "01/25/1996"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "09/15/1986"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "32"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Bargbo"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Alberto Raya"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Meti"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Abiodun"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "10/24/2010"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "07/26/1989"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "8"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Majihun MCHP"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Unassigned"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Eshe"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Okeke"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "01/31/1995"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "01/31/1995"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "63"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Mambiama CHP"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Shadrias Pearson"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Obi"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Okafor"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "06/07/1990"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "01/03/2006"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "28"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Dalakuru CHP"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Anatoliy Shcherbatykh"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Incomplete")))));

const FixedHeader = FixedHeaderTemplate.bind({});
exports.FixedHeader = FixedHeader;
FixedHeader.args = {
  borderless: true
};

const FixedFirstColumnTemplate = args => /*#__PURE__*/_react.default.createElement(_tableScrollBox.TableScrollBox, {
  maxWidth: "500px"
}, /*#__PURE__*/_react.default.createElement(_table.Table, args, /*#__PURE__*/_react.default.createElement(_tableHead.TableHead, null, /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0",
  left: "0"
}, "First name"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, null, "Last name"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, null, "Incident date"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, null, "Last updated"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, null, "Age"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, null, "Registering unit"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, null, "Assigned user"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, null, "Status"))), /*#__PURE__*/_react.default.createElement(_tableBody.TableBody, null, /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0"
}, "Onyekachukwu"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Kariuki"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "05/25/1972"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "66"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Jawi"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Sofie Hubert"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0"
}, "Kwasi"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Okafor"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "08/11/2010"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "02/26/1991"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "38"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Mokassie MCHP"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Dashonte Clarke"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0"
}, "Siyabonga"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Abiodun"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "07/21/1981"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "98"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Bathurst MCHP"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Unassigned"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0"
}, "Chiyembekezo"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Okeke"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "01/23/1982"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "07/15/2003"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "2"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Mayolla MCHP"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Wan Gengxin"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0"
}, "Mtendere"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Afolayan"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "08/12/1994"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "05/12/1972"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "37"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Gbangadu MCHP"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Gvozden Boskovsky"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0"
}, "Inyene"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Okonkwo"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "04/01/1971"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "03/16/2000"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "70"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Kunike Barina"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Oscar de la Cavaller\xEDa"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0"
}, "Amaka"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Pretorius"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "01/25/1996"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "09/15/1986"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "32"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Bargbo"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Alberto Raya"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0"
}, "Meti"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Abiodun"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "10/24/2010"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "07/26/1989"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "8"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Majihun MCHP"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Unassigned"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0"
}, "Eshe"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Okeke"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "01/31/1995"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "01/31/1995"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "63"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Mambiama CHP"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Shadrias Pearson"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0"
}, "Obi"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Okafor"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "06/07/1990"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "01/03/2006"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "28"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Dalakuru CHP"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Anatoliy Shcherbatykh"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Incomplete")))));

const FixedFirstColumn = FixedFirstColumnTemplate.bind({});
exports.FixedFirstColumn = FixedFirstColumn;
FixedFirstColumn.args = {
  borderless: true,
  width: '1000px'
};

const FixedHeaderAndTwoColumnsTemplate = args => /*#__PURE__*/_react.default.createElement(_tableScrollBox.TableScrollBox, {
  maxHeight: "350px",
  maxWidth: "500px"
}, /*#__PURE__*/_react.default.createElement(_table.Table, args, /*#__PURE__*/_react.default.createElement(_tableHead.TableHead, null, /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0",
  left: "0",
  width: "120px"
}, "First name"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0",
  left: "120px"
}, "Last name"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Incident date"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Last updated"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Age"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Registering unit"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Assigned user"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Status"))), /*#__PURE__*/_react.default.createElement(_tableBody.TableBody, null, /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Onyekachukwu"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Kariuki"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "05/25/1972"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "66"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Jawi"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Sofie Hubert"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Kwasi"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Okafor"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "08/11/2010"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "02/26/1991"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "38"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Mokassie MCHP"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Dashonte Clarke"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Siyabonga"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Abiodun"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "07/21/1981"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "98"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Bathurst MCHP"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Unassigned"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Chiyembekezo"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Okeke"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "01/23/1982"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "07/15/2003"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "2"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Mayolla MCHP"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Wan Gengxin"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Mtendere"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Afolayan"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "08/12/1994"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "05/12/1972"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "37"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Gbangadu MCHP"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Gvozden Boskovsky"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Inyene"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Okonkwo"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "04/01/1971"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "03/16/2000"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "70"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Kunike Barina"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Oscar de la Cavaller\xEDa"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Amaka"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Pretorius"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "01/25/1996"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "09/15/1986"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "32"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Bargbo"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Alberto Raya"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Meti"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Abiodun"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "10/24/2010"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "07/26/1989"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "8"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Majihun MCHP"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Unassigned"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Eshe"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Okeke"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "01/31/1995"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "01/31/1995"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "63"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Mambiama CHP"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Shadrias Pearson"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Obi"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Okafor"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "06/07/1990"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "01/03/2006"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "28"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Dalakuru CHP"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Anatoliy Shcherbatykh"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Incomplete")))));

const FixedHeaderAndTwoColumns = FixedHeaderAndTwoColumnsTemplate.bind({});
exports.FixedHeaderAndTwoColumns = FixedHeaderAndTwoColumns;
FixedHeaderAndTwoColumns.args = {
  layout: 'fixed',
  borderless: true,
  width: '1000px'
};

const ScrollingTableWithToolbarsTemplate = args => /*#__PURE__*/_react.default.createElement(_uiCore.Box, {
  width: "500px"
}, /*#__PURE__*/_react.default.createElement(_tableToolbar.TableToolbar, null, /*#__PURE__*/_react.default.createElement("p", null, "Content")), /*#__PURE__*/_react.default.createElement(_tableScrollBox.TableScrollBox, {
  maxHeight: "350px",
  maxWidth: "500px"
}, /*#__PURE__*/_react.default.createElement(_table.Table, args, /*#__PURE__*/_react.default.createElement(_tableHead.TableHead, null, /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0",
  left: "0",
  width: "120px"
}, "First name"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0",
  left: "120px"
}, "Last name"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Incident date"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Last updated"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Age"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Registering unit"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Assigned user"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  top: "0"
}, "Status"))), /*#__PURE__*/_react.default.createElement(_tableBody.TableBody, null, /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Onyekachukwu"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Kariuki"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "05/25/1972"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "66"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Jawi"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Sofie Hubert"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Kwasi"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Okafor"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "08/11/2010"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "02/26/1991"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "38"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Mokassie MCHP"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Dashonte Clarke"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Siyabonga"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Abiodun"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "07/21/1981"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "02/06/2007"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "98"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Bathurst MCHP"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Unassigned"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Chiyembekezo"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Okeke"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "01/23/1982"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "07/15/2003"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "2"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Mayolla MCHP"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Wan Gengxin"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Mtendere"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Afolayan"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "08/12/1994"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "05/12/1972"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "37"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Gbangadu MCHP"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Gvozden Boskovsky"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Inyene"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Okonkwo"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "04/01/1971"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "03/16/2000"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "70"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Kunike Barina"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Oscar de la Cavaller\xEDa"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Amaka"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Pretorius"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "01/25/1996"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "09/15/1986"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "32"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Bargbo"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Alberto Raya"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Meti"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Abiodun"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "10/24/2010"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "07/26/1989"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "8"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Majihun MCHP"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Unassigned"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Complete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Eshe"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Okeke"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "01/31/1995"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "01/31/1995"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "63"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Mambiama CHP"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Shadrias Pearson"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Incomplete")), /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "0",
  width: "120px"
}, "Obi"), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, {
  fixed: true,
  left: "120px"
}, "Okafor"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "06/07/1990"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "01/03/2006"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "28"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Dalakuru CHP"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Anatoliy Shcherbatykh"), /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, null, "Incomplete"))))), /*#__PURE__*/_react.default.createElement(_tableToolbar.TableToolbar, {
  position: "bottom"
}, /*#__PURE__*/_react.default.createElement("p", null, "Content (bottom)")));

const ScrollingTableWithToolbars = ScrollingTableWithToolbarsTemplate.bind({});
exports.ScrollingTableWithToolbars = ScrollingTableWithToolbars;
ScrollingTableWithToolbars.args = {
  layout: 'fixed',
  borderless: true,
  width: '1000px'
};

const HeaderActionsTemplate = args => /*#__PURE__*/_react.default.createElement(_table.Table, args, /*#__PURE__*/_react.default.createElement(_tableHead.TableHead, null, /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, null, /*#__PURE__*/_react.default.createElement("span", {
  style: {
    display: 'flex',
    height: '100%',
    alignItems: 'center'
  }
}, "First name", /*#__PURE__*/_react.default.createElement(_tableHeaderCellAction.TableHeaderCellAction, null, /*#__PURE__*/_react.default.createElement(_uiIcons.IconAttachment16, {
  color: "green"
})))), /*#__PURE__*/_react.default.createElement(_tableHeaderCell.TableHeaderCell, null, /*#__PURE__*/_react.default.createElement("span", {
  style: {
    display: 'flex',
    height: '100%',
    alignItems: 'center'
  }
}, "Last name", /*#__PURE__*/_react.default.createElement(_tableHeaderCellAction.TableHeaderCellAction, null, /*#__PURE__*/_react.default.createElement(_uiIcons.IconAttachment16, {
  color: "red"
})))))), /*#__PURE__*/_react.default.createElement(_tableBody.TableBody, null, /*#__PURE__*/_react.default.createElement(_tableRow.TableRow, null, /*#__PURE__*/_react.default.createElement(_tableDataCell.TableDataCell, {
  colSpan: "3"
}))));

const HeaderActions = HeaderActionsTemplate.bind({});
exports.HeaderActions = HeaderActions;
HeaderActions.args = {};
