// @dhis2-ui/table/build/cjs/data-table/table-elements/table.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Table = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _uiConstants = require("@dhis2/ui-constants");

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const Table = /*#__PURE__*/(0, _react.forwardRef)(({
  children,
  className,
  dataTest,
  layout,
  role,
  borderless,
  width,
  ...props
}, ref) => /*#__PURE__*/_react.default.createElement("table", _extends({}, props, {
  "data-test": dataTest,
  ref: ref,
  role: role,
  className: _style.default.dynamic([["3140496910", [layout, width, _uiConstants.colors.grey300]]]) + " " + ((0, _classnames.default)(className, {
    borderless
  }) || "")
}), children, /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "3140496910",
  dynamic: [layout, width, _uiConstants.colors.grey300]
}, [`table.__jsx-style-dynamic-selector{table-layout:${layout};border-collapse:separate;border-spacing:0;width:${width};box-sizing:border-box;border:1px solid ${_uiConstants.colors.grey300};}`, "table.borderless.__jsx-style-dynamic-selector{border:none;}"])));
exports.Table = Table;
Table.displayName = 'Table';
Table.defaultProps = {
  dataTest: 'dhis2-uicore-table',
  width: '100%',
  layout: 'auto'
};
Table.propTypes = {
  /**
   * Removes border from the table
   */
  borderless: _propTypes.default.bool,

  /**
   * Should be `<TableHead>`, `<TableBody>`, and `<TableFoot>` components
   */
  children: _propTypes.default.node,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string,

  /**
   * Sets the `table-layout` property. Switching to `fixed` can prevent style
   * issues when dealing with a table with multiple frozen columns or when dealing
   * with filter elements in the table headers.
   */
  layout: _propTypes.default.oneOf(['auto', 'fixed', 'initial', 'inherit']),
  role: _propTypes.default.string,

  /**
   * Sets the `width` property. Providing an explicit width can prevent style
   * issues when dealing with horizontally scrolling tables with a fixed layout.
   */
  width: _propTypes.default.string
};
