// @dhis2-ui/table/build/cjs/data-table/table-elements/table-header-cell-action.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TableHeaderCellAction = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _uiConstants = require("@dhis2/ui-constants");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const TableHeaderCellAction = /*#__PURE__*/(0, _react.forwardRef)(({
  onClick,
  children,
  className,
  dataTest,
  title,
  ...props
}, ref) => /*#__PURE__*/_react.default.createElement("button", _extends({}, props, {
  "data-test": dataTest,
  onClick: onClick,
  title: title,
  ref: ref,
  className: _style.default.dynamic([["1702694916", [_uiConstants.colors.grey400]]]) + " " + (className || "")
}), children, /*#__PURE__*/_react.default.createElement(_style.default, {
  id: "1702694916",
  dynamic: [_uiConstants.colors.grey400]
}, ["button.__jsx-style-dynamic-selector{border:none;padding:0;background:transparent;width:24px;height:100%;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;cursor:pointer;}", `button.__jsx-style-dynamic-selector:hover,button.__jsx-style-dynamic-selector:focus-visible{background:${_uiConstants.colors.grey400};}`, "button.__jsx-style-dynamic-selector:active,button.__jsx-style-dynamic-selector:focus{outline:none;}"])));
exports.TableHeaderCellAction = TableHeaderCellAction;
TableHeaderCellAction.displayName = 'TableHeaderCellAction';
TableHeaderCellAction.defaultProps = {
  dataTest: 'dhis2-uicore-tableheadercellaction'
};
TableHeaderCellAction.propTypes = {
  children: _propTypes.default.node,
  className: _propTypes.default.string,
  dataTest: _propTypes.default.string,
  title: _propTypes.default.string,
  onClick: _propTypes.default.func
};
