// @dhis2-ui/alert/build/cjs/alert-bar/actions.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.actionsPropType = exports.Actions = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _propTypes = _interopRequireDefault(require("@dhis2/prop-types"));

var _uiConstants = require("@dhis2/ui-constants");

var _react = _interopRequireDefault(require("react"));

var _action = require("./action.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const Actions = ({
  actions,
  hide,
  dataTest
}) => {
  if (!actions) {
    return null;
  }

  return /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["961263490", [_uiConstants.spacers.dp48, _uiConstants.spacers.dp12]]])
  }, actions.map(action => /*#__PURE__*/_react.default.createElement(_action.Action, _extends({
    key: action.label
  }, action, {
    hide: hide,
    dataTest: `${dataTest}-action`
  }))), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "961263490",
    dynamic: [_uiConstants.spacers.dp48, _uiConstants.spacers.dp12]
  }, [`div.__jsx-style-dynamic-selector{margin-left:${_uiConstants.spacers.dp48};margin-right:-${_uiConstants.spacers.dp12};}`]));
};

exports.Actions = Actions;

const actionsPropType = _propTypes.default.arrayWithLength(0, 2, _propTypes.default.shape({
  label: _propTypes.default.string.isRequired,
  onClick: _propTypes.default.func.isRequired
}));

exports.actionsPropType = actionsPropType;
Actions.propTypes = {
  dataTest: _propTypes.default.string.isRequired,
  hide: _propTypes.default.func.isRequired,
  actions: actionsPropType
};
