// @dhis2-ui/alert/build/cjs/alert-bar/alert-bar.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.HiddenProp = exports.TextOverflow = exports.Icons = exports.WithActions = exports.AutoHiding = exports.States = exports.Default = exports.default = void 0;

var _uiIcons = require("@dhis2/ui-icons");

var _react = _interopRequireWildcard(require("react"));

var _index = require("./index.js");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

const subtitle = `
A floating alert that informs the user about temporary information
in the context of the current screen.
`;
const description = `
Alert bars notify a user of some information. There are different types of
alert bar for displaying different types of content. Use the alert bar type
that matches your content type and importance. Note that alert bar can be
ignored by the user, so they shouldn't be used for content that needs to
block an app flow, use a modal instead.

Alert bars are always displayed at centered and fixed at the bottom of the
screen. Some types of alert bar dismiss after a set time, others must be
dismissed by the user.

See specification: [Design System](https://github.com/dhis2/design-system/blob/master/molecules/alertbar.md)

\`\`\`js
import { AlertBar } from '@dhis2/ui'
\`\`\`
`;

const Wrapper = fn => /*#__PURE__*/_react.default.createElement("div", {
  style: {
    height: '260px'
  }
}, /*#__PURE__*/_react.default.createElement("div", {
  className: "alert-bars",
  style: {
    width: '100%',
    position: 'fixed',
    bottom: 0,
    left: 0,
    paddingLeft: 16
  }
}, fn()));

const alertTypeArgType = {
  table: {
    type: {
      summary: 'bool',
      detail: "'success', 'warning', and 'critical' are mutually exclusive props"
    }
  },
  control: {
    type: 'boolean'
  }
};
const iconArgType = {
  table: {
    type: {
      summary: 'bool | element'
    }
  }
};
const actionsArgType = {
  table: {
    type: {
      summary: '[{ label: string, onClick: func }]'
    }
  }
};
var _default = {
  title: 'Feedback/Alerts/Alert Bar',
  component: _index.AlertBar,
  decorators: [Wrapper],
  parameters: {
    componentSubtitle: subtitle,
    docs: {
      description: {
        component: description
      }
    }
  },
  argTypes: {
    actions: { ...actionsArgType
    },
    critical: { ...alertTypeArgType
    },
    success: { ...alertTypeArgType
    },
    warning: { ...alertTypeArgType
    },
    icon: { ...iconArgType
    }
  }
};
exports.default = _default;

const Default = args => /*#__PURE__*/_react.default.createElement(_index.AlertBar, args, "Default - I will autohide");

exports.Default = Default;

const States = () => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
  permanent: true
}, "Default (info)"), /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
  permanent: true,
  success: true
}, "Success"), /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
  permanent: true,
  warning: true
}, "Warning"), /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
  permanent: true,
  critical: true
}, "Critical"));

exports.States = States;

const AutoHiding = () => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
  permanent: true
}, "Permanent never auto-hides"), /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
  warning: true
}, "Warning never auto-hides"), /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
  critical: true
}, "Critial never auto-hides"), /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
  duration: 2000,
  onHidden: (payload, event) => {
    console.log('onHidden payload', payload);
    console.log('onHidden event', event);
  }
}, "Custom duration, hides after 2s"), /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
  onHidden: (payload, event) => {
    console.log('onHidden payload', payload);
    console.log('onHidden event', event);
  }
}, "Default auto-hides after 8s"));

exports.AutoHiding = AutoHiding;
AutoHiding.storyName = 'Auto hiding';

const WithActions = () => /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
  permanent: true,
  actions: [{
    label: 'Save',
    onClick: () => {}
  }, {
    label: 'Cancel',
    onClick: () => {}
  }]
}, "With Actions");

exports.WithActions = WithActions;
WithActions.storyName = 'With actions';

const Icons = () => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
  permanent: true
}, "Default icon"), /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
  permanent: true,
  icon: false
}, "No icon"), /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
  permanent: true,
  icon: /*#__PURE__*/_react.default.createElement(_uiIcons.IconFile16, null)
}, "Custom icon"));

exports.Icons = Icons;

const TextOverflow = () => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
  permanent: true
}, "Short text"), /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
  permanent: true
}, "If the alert bar gets a ver long text, it will grow to a maximum of 600px and the text will overflow across several lines. If there are multiple AlertBars in a stack, they will all grow to the size of the widest sibling."));

exports.TextOverflow = TextOverflow;
TextOverflow.storyName = 'Text overflow';

const HiddenProp = () => {
  const [hidden, setHidden] = (0, _react.useState)(true);

  const toggleVisibility = () => setHidden(prevHidden => !prevHidden);

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("button", {
    style: {
      display: 'block',
      position: 'fixed',
      bottom: 150,
      left: 10
    },
    onClick: toggleVisibility
  }, hidden ? 'Show' : 'Hide'), /*#__PURE__*/_react.default.createElement(_index.AlertBar, {
    permanent: true,
    hidden: hidden
  }, "Short text"));
};

exports.HiddenProp = HiddenProp;
HiddenProp.storyName = 'Hidden prop';
