// @dhis2-ui/box/build/cjs/box.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Overflow = exports.MaxWidth = exports.MinWidth = exports.Width = exports.MinHeight = exports.MaxHeight = exports.Height = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _box = require("./box.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const description = `
A box for creating some layout on the page.

\`\`\`js
import { Box } from '@dhis2/ui'
\`\`\`
`;
var _default = {
  title: 'Layout/Box',
  component: _box.Box,
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  }
};
exports.default = _default;

const Template = args => /*#__PURE__*/_react.default.createElement(_box.Box, args);

const Default = Template.bind({});
exports.Default = Default;
Default.args = {
  children: 'I am a child in a Box.'
};
const Height = Template.bind({});
exports.Height = Height;
Height.args = { ...Default.args,
  height: '250px'
};
const MaxHeight = Template.bind({});
exports.MaxHeight = MaxHeight;
MaxHeight.args = {
  children: /*#__PURE__*/_react.default.createElement("p", {
    style: {
      height: '500px'
    }
  }, "I am a tall child in a low Box."),
  maxHeight: '250px'
};
const MinHeight = Template.bind({});
exports.MinHeight = MinHeight;
MinHeight.args = { ...Default.args,
  minHeight: '50vh'
};
const Width = Template.bind({});
exports.Width = Width;
Width.args = { ...Default.args,
  width: '250px'
};
const MinWidth = Template.bind({});
exports.MinWidth = MinWidth;
MinWidth.args = { ...Default.args,
  minWidth: '50vh'
};
const MaxWidth = Template.bind({});
exports.MaxWidth = MaxWidth;
MaxWidth.args = { ...Default.args,
  maxWidth: '50vh'
};
const Overflow = Template.bind({});
exports.Overflow = Overflow;
Overflow.args = {
  maxHeight: '250px',
  overflow: 'scroll',
  children: /*#__PURE__*/_react.default.createElement("p", {
    style: {
      height: '500px'
    }
  }, "I am a tall child in a low Box, and my parent clips me")
};
