// @dhis2-ui/checkbox/build/cjs/checkbox/checkbox.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ImageLabelDense = exports.ErrorDense = exports.WarningDense = exports.ValidDense = exports.DisabledDense = exports.IndeterminateDense = exports.CheckedDense = exports.FocusedCheckedDense = exports.FocusedUncheckedDense = exports.DefaultDense = exports.ImageLabel = exports.Error = exports.Warning = exports.Valid = exports.Disabled = exports.Indeterminate = exports.Checked = exports.FocusedChecked = exports.FocusedUnchecked = exports.Default = exports.default = void 0;

var _uiConstants = require("@dhis2/ui-constants");

var _react = _interopRequireDefault(require("react"));

var _index = require("./index.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const subtitle = 'A checkbox is a control that allows a user to toggle an option.';
const description = `
Checkboxes are used when an option can be toggled on or off. Toggling a checkbox on (true) is always considered a positive action and should reflect a positive/true/on state. Multiple checkboxes can be used in a list where a user can toggle multiple elements.

Do not use checkboxes in a list of several options where only a single option can be toggled, use [radio buttons](../?path=/docs/forms-radio-radio--default) here instead.

If there are many options that need to select from, consider using a [select](../?path=/docs/forms-single-select-single-select--with-options) instead.

\`\`\`js
import { Checkbox } from '@dhis2/ui'
\`\`\`
`;

window.onChange = (payload, event) => {
  console.log('onClick payload', payload);
  console.log('onClick event', event);
};

window.onFocus = (payload, event) => {
  console.log('onFocus payload', payload);
  console.log('onFocus event', event);
};

window.onBlur = (payload, event) => {
  console.log('onBlur payload', payload);
  console.log('onBlur event', event);
};

const onChange = (...args) => window.onChange(...args);

const onFocus = (...args) => window.onFocus(...args);

const onBlur = (...args) => window.onBlur(...args);

const defaultArgs = {
  name: 'Ex',
  label: 'Checkbox',
  onChange,
  onFocus,
  onBlur
};
const uniqueOnStateArgType = {
  table: {
    type: {
      summary: 'bool',
      detail: "'checked' and 'indeterminate' are mutually exclusive props"
    }
  },
  control: {
    type: 'boolean'
  }
};
var _default = {
  title: 'Forms/Checkbox/Checkbox',
  component: _index.Checkbox,
  parameters: {
    componentSubtitle: subtitle,
    docs: {
      description: {
        component: description
      }
    }
  },
  // Sets default args on all stories unless overridden
  args: { ...defaultArgs
  },
  argTypes: {
    checked: { ...uniqueOnStateArgType
    },
    indeterminate: { ...uniqueOnStateArgType
    },
    valid: { ..._uiConstants.sharedPropTypes.statusArgType
    },
    warning: { ..._uiConstants.sharedPropTypes.statusArgType
    },
    error: { ..._uiConstants.sharedPropTypes.statusArgType
    }
  }
};
exports.default = _default;

const Template = args => /*#__PURE__*/_react.default.createElement(_index.Checkbox, args);

const CheckedUncheckedTemplate = args => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_index.Checkbox, args), /*#__PURE__*/_react.default.createElement(_index.Checkbox, _extends({
  checked: true
}, args)));

const Default = Template.bind({});
exports.Default = Default;
Default.args = {
  value: 'default'
};

const FocusedUnchecked = args => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_index.Checkbox, _extends({
  initialFocus: true,
  value: "default",
  className: "initially-focused"
}, args)), /*#__PURE__*/_react.default.createElement(_index.Checkbox, _extends({
  value: "default2",
  className: "initially-unfocused"
}, args)));

exports.FocusedUnchecked = FocusedUnchecked;
FocusedUnchecked.storyName = 'Focused unchecked';
/**
 * 'initialFocus' causes docs page to scroll away every time a control is
 * changed, so it's disabled
 */

FocusedUnchecked.parameters = {
  docs: {
    disable: true
  }
};

const FocusedChecked = args => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_index.Checkbox, _extends({
  initialFocus: true,
  checked: true,
  value: "default"
}, args)), /*#__PURE__*/_react.default.createElement(_index.Checkbox, _extends({
  checked: true,
  value: "default2"
}, args)));

exports.FocusedChecked = FocusedChecked;
FocusedChecked.storyName = 'Focused checked';
FocusedChecked.parameters = {
  docs: {
    disable: true
  }
};
const Checked = Template.bind({});
exports.Checked = Checked;
Checked.args = {
  checked: true,
  value: 'checked'
};
const Indeterminate = Template.bind({});
exports.Indeterminate = Indeterminate;
Indeterminate.args = {
  indeterminate: true,
  value: 'checked'
};
const Disabled = CheckedUncheckedTemplate.bind({});
exports.Disabled = Disabled;
Disabled.args = {
  disabled: true,
  value: 'disabled'
};
const Valid = CheckedUncheckedTemplate.bind({});
exports.Valid = Valid;
Valid.args = {
  valid: true,
  value: 'valid'
};
const Warning = CheckedUncheckedTemplate.bind({});
exports.Warning = Warning;
Warning.args = {
  warning: true,
  value: 'warning'
};
const Error = CheckedUncheckedTemplate.bind({});
exports.Error = Error;
Error.args = {
  error: true,
  value: 'error'
};
const ImageLabel = Template.bind({});
exports.ImageLabel = ImageLabel;
ImageLabel.args = {
  label: /*#__PURE__*/_react.default.createElement("img", {
    src: "https://picsum.photos/id/82/200/100"
  }),
  value: 'with-help'
};
const DefaultDense = Template.bind({});
exports.DefaultDense = DefaultDense;
DefaultDense.args = {
  dense: true,
  value: 'default'
};
DefaultDense.storyName = 'Default - Dense';
const FocusedUncheckedDense = Template.bind({});
exports.FocusedUncheckedDense = FocusedUncheckedDense;
FocusedUncheckedDense.args = {
  dense: true,
  initialFocus: true,
  value: 'default'
};
FocusedUncheckedDense.storyName = 'Focused unchecked - Dense';
FocusedUncheckedDense.parameters = {
  docs: {
    disable: true
  }
};
const FocusedCheckedDense = Template.bind({});
exports.FocusedCheckedDense = FocusedCheckedDense;
FocusedCheckedDense.args = { ...FocusedUncheckedDense.args,
  checked: true
};
FocusedCheckedDense.storyName = 'Focused checked - Dense';
FocusedCheckedDense.parameters = {
  docs: {
    disable: true
  }
};
const CheckedDense = Template.bind({});
exports.CheckedDense = CheckedDense;
CheckedDense.args = {
  dense: true,
  checked: true,
  value: 'checked'
};
CheckedDense.storyName = 'Checked - Dense';
const IndeterminateDense = Template.bind({});
exports.IndeterminateDense = IndeterminateDense;
IndeterminateDense.args = {
  dense: true,
  indeterminate: true,
  value: 'checked'
};
IndeterminateDense.storyName = 'Indeterminate - Dense';
const DisabledDense = CheckedUncheckedTemplate.bind({});
exports.DisabledDense = DisabledDense;
DisabledDense.args = { ...Disabled.args,
  dense: true
};
DisabledDense.storyName = 'Disabled - Dense';
const ValidDense = CheckedUncheckedTemplate.bind({});
exports.ValidDense = ValidDense;
ValidDense.args = { ...Valid.args,
  dense: true
};
ValidDense.storyName = 'Valid - Dense';
const WarningDense = CheckedUncheckedTemplate.bind({});
exports.WarningDense = WarningDense;
WarningDense.args = { ...Warning.args,
  dense: true
};
WarningDense.storyName = 'Warning - Dense';
const ErrorDense = CheckedUncheckedTemplate.bind({});
exports.ErrorDense = ErrorDense;
ErrorDense.args = { ...Error.args,
  dense: true
};
ErrorDense.storyName = 'Error - Dense';
const ImageLabelDense = Template.bind({});
exports.ImageLabelDense = ImageLabelDense;
ImageLabelDense.args = { ...ImageLabel.args,
  dense: true
};
ImageLabelDense.storyName = 'Image label - Dense';
