// @dhis2-ui/menu/build/cjs/menu-item/menu-item.stories.e2e.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WithIcon = exports.WithTarget = exports.WithHref = exports.WithOnClickAndValue = exports.WithLabel = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _index = require("../index.js");

var _menuItem = require("./menu-item.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

window.onClick = window.Cypress && window.Cypress.cy.stub();
var _default = {
  title: 'MenuItem',
  component: _menuItem.MenuItem,
  decorators: [story => /*#__PURE__*/_react.default.createElement(_index.Menu, null, story())]
};
exports.default = _default;

const WithLabel = () => /*#__PURE__*/_react.default.createElement(_menuItem.MenuItem, {
  label: "label"
});

exports.WithLabel = WithLabel;

const WithOnClickAndValue = () => /*#__PURE__*/_react.default.createElement(_menuItem.MenuItem, {
  value: "Value",
  onClick: window.onClick,
  label: "Menu item"
});

exports.WithOnClickAndValue = WithOnClickAndValue;

const WithHref = () => /*#__PURE__*/_react.default.createElement(_menuItem.MenuItem, {
  href: "url.test",
  label: "Menu item"
});

exports.WithHref = WithHref;

const WithTarget = () => /*#__PURE__*/_react.default.createElement(_menuItem.MenuItem, {
  href: "url.test",
  target: "_blank",
  label: "Menu item"
});

exports.WithTarget = WithTarget;

const WithIcon = () => /*#__PURE__*/_react.default.createElement(_menuItem.MenuItem, {
  icon: /*#__PURE__*/_react.default.createElement("div", null, "Icon"),
  label: "Menu item"
});

exports.WithIcon = WithIcon;
