// @dhis2-ui/modal/build/es/modal/modal.js
import _JSXStyle from "styled-jsx/style";
import { Card } from '@dhis2-ui/card';
import { Center } from '@dhis2-ui/center';
import { Layer } from '@dhis2-ui/layer';
import { spacers, spacersNum, sharedPropTypes } from '@dhis2/ui-constants';
import cx from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
const scrollBoxCard = {
  styles: /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "2814715826"
  }, [`div.jsx-2814715826{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;max-height:calc(100vh - ${2 * spacersNum.dp64}px);max-width:calc(100vw - ${2 * spacersNum.dp64}px);}`]),
  className: "jsx-2814715826"
};
const centeredContent = {
  styles: /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "1378205604"
  }, [`.top.jsx-1378205604{padding-top:${spacers.dp64};}`, `.bottom.jsx-1378205604{padding-bottom:${spacers.dp64};}`]),
  className: "jsx-1378205604"
};
const layerStyles = {
  styles: /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "427720588"
  }, ["div.jsx-427720588{display:none;}"]),
  className: "jsx-427720588"
};
export const Modal = ({
  children,
  className,
  dataTest,
  hide,
  large,
  onClose,
  position,
  small
}) => {
  return /*#__PURE__*/React.createElement(Layer, {
    onClick: onClose,
    className: hide ? layerStyles.className : '',
    translucent: !hide
  }, /*#__PURE__*/React.createElement(Center, {
    position: position,
    className: centeredContent.className
  }, /*#__PURE__*/React.createElement("aside", {
    role: "dialog",
    "aria-modal": "true",
    "data-test": dataTest,
    className: _JSXStyle.dynamic([["2700832039", [2 * spacersNum.dp64, 2 * spacersNum.dp64]]]) + " " + (cx(className, {
      small,
      large
    }) || "")
  }, /*#__PURE__*/React.createElement(Card, {
    className: scrollBoxCard.className
  }, children)), scrollBoxCard.styles, layerStyles.styles, centeredContent.styles), /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "2700832039",
    dynamic: [2 * spacersNum.dp64, 2 * spacersNum.dp64]
  }, [`aside.__jsx-style-dynamic-selector{overflow-y:hidden;height:auto;max-height:calc(100vh - ${2 * spacersNum.dp64}px);max-width:calc(100vw - ${2 * spacersNum.dp64}px);width:600px;}`, "aside.small.__jsx-style-dynamic-selector{width:400px;}", "aside.large.__jsx-style-dynamic-selector{width:800px;}"]));
};
Modal.defaultProps = {
  dataTest: 'dhis2-uicore-modal',
  position: 'top'
};
Modal.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  dataTest: PropTypes.string,
  hide: PropTypes.bool,
  large: sharedPropTypes.sizePropType,
  position: sharedPropTypes.insideAlignmentPropType,
  small: sharedPropTypes.sizePropType,

  /** Callback used when screen cover is clicked */
  onClose: PropTypes.func
};
